// @aws-sdk/client-codebuild/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "codebuild",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class CodeBuildClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultCodeBuildHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class CodeBuildServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, CodeBuildServiceException.prototype);
    }
}

class AccountLimitExceededException extends CodeBuildServiceException {
    name = "AccountLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AccountLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccountLimitExceededException.prototype);
    }
}
class AccountSuspendedException extends CodeBuildServiceException {
    name = "AccountSuspendedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AccountSuspendedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccountSuspendedException.prototype);
    }
}
class InvalidInputException extends CodeBuildServiceException {
    name = "InvalidInputException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidInputException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidInputException.prototype);
    }
}
class ResourceAlreadyExistsException extends CodeBuildServiceException {
    name = "ResourceAlreadyExistsException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceAlreadyExistsException.prototype);
    }
}
class OAuthProviderException extends CodeBuildServiceException {
    name = "OAuthProviderException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "OAuthProviderException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OAuthProviderException.prototype);
    }
}
class ResourceNotFoundException extends CodeBuildServiceException {
    name = "ResourceNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
}

const _ALEE = "AccountLimitExceededException";
const _ARC = "AutoRetryConfig";
const _ASE = "AccountSuspendedException";
const _B = "Build";
const _BA = "BuildArtifacts";
const _BAL = "BuildArtifactsList";
const _BB = "BuildBatch";
const _BBF = "BuildBatchFilter";
const _BBP = "BuildBatchPhase";
const _BBPu = "BuildBatchPhases";
const _BBu = "BuildBatches";
const _BDB = "BatchDeleteBuilds";
const _BDBI = "BatchDeleteBuildsInput";
const _BDBO = "BatchDeleteBuildsOutput";
const _BG = "BuildGroup";
const _BGB = "BatchGetBuilds";
const _BGBB = "BatchGetBuildBatches";
const _BGBBI = "BatchGetBuildBatchesInput";
const _BGBBO = "BatchGetBuildBatchesOutput";
const _BGBI = "BatchGetBuildsInput";
const _BGBO = "BatchGetBuildsOutput";
const _BGCE = "BatchGetCommandExecutions";
const _BGCEI = "BatchGetCommandExecutionsInput";
const _BGCEO = "BatchGetCommandExecutionsOutput";
const _BGF = "BatchGetFleets";
const _BGFI = "BatchGetFleetsInput";
const _BGFO = "BatchGetFleetsOutput";
const _BGP = "BatchGetProjects";
const _BGPI = "BatchGetProjectsInput";
const _BGPO = "BatchGetProjectsOutput";
const _BGR = "BatchGetReports";
const _BGRG = "BatchGetReportGroups";
const _BGRGI = "BatchGetReportGroupsInput";
const _BGRGO = "BatchGetReportGroupsOutput";
const _BGRI = "BatchGetReportsInput";
const _BGRO = "BatchGetReportsOutput";
const _BGS = "BatchGetSandboxes";
const _BGSI = "BatchGetSandboxesInput";
const _BGSO = "BatchGetSandboxesOutput";
const _BGu = "BuildGroups";
const _BND = "BuildNotDeleted";
const _BNDu = "BuildsNotDeleted";
const _BP = "BuildPhase";
const _BPu = "BuildPhases";
const _BR = "BatchRestrictions";
const _BS = "BuildSummary";
const _BSC = "BuildStatusConfig";
const _BSu = "BuildSummaries";
const _Bu = "Builds";
const _CC = "CodeCoverage";
const _CCRS = "CodeCoverageReportSummary";
const _CCo = "ComputeConfiguration";
const _CCod = "CodeCoverages";
const _CE = "CommandExecution";
const _CEo = "CommandExecutions";
const _CF = "CreateFleet";
const _CFI = "CreateFleetInput";
const _CFO = "CreateFleetOutput";
const _CP = "CreateProject";
const _CPI = "CreateProjectInput";
const _CPO = "CreateProjectOutput";
const _CRG = "CreateReportGroup";
const _CRGI = "CreateReportGroupInput";
const _CRGO = "CreateReportGroupOutput";
const _CW = "CreateWebhook";
const _CWI = "CreateWebhookInput";
const _CWLC = "CloudWatchLogsConfig";
const _CWO = "CreateWebhookOutput";
const _DBB = "DeleteBuildBatch";
const _DBBI = "DeleteBuildBatchInput";
const _DBBO = "DeleteBuildBatchOutput";
const _DCC = "DescribeCodeCoverages";
const _DCCI = "DescribeCodeCoveragesInput";
const _DCCO = "DescribeCodeCoveragesOutput";
const _DF = "DeleteFleet";
const _DFI = "DeleteFleetInput";
const _DFO = "DeleteFleetOutput";
const _DP = "DeleteProject";
const _DPI = "DeleteProjectInput";
const _DPO = "DeleteProjectOutput";
const _DR = "DeleteReport";
const _DRG = "DeleteReportGroup";
const _DRGI = "DeleteReportGroupInput";
const _DRGO = "DeleteReportGroupOutput";
const _DRI = "DeleteReportInput";
const _DRO = "DeleteReportOutput";
const _DRP = "DeleteResourcePolicy";
const _DRPI = "DeleteResourcePolicyInput";
const _DRPO = "DeleteResourcePolicyOutput";
const _DS = "DebugSession";
const _DSC = "DeleteSourceCredentials";
const _DSCI = "DeleteSourceCredentialsInput";
const _DSCO = "DeleteSourceCredentialsOutput";
const _DSS = "DockerServerStatus";
const _DSo = "DockerServer";
const _DTC = "DescribeTestCases";
const _DTCI = "DescribeTestCasesInput";
const _DTCO = "DescribeTestCasesOutput";
const _DW = "DeleteWebhook";
const _DWI = "DeleteWebhookInput";
const _DWO = "DeleteWebhookOutput";
const _EEV = "ExportedEnvironmentVariable";
const _EEVx = "ExportedEnvironmentVariables";
const _EI = "EnvironmentImage";
const _EIn = "EnvironmentImages";
const _EL = "EnvironmentLanguage";
const _ELn = "EnvironmentLanguages";
const _EP = "EnvironmentPlatform";
const _EPn = "EnvironmentPlatforms";
const _EV = "EnvironmentVariable";
const _EVn = "EnvironmentVariables";
const _F = "Fleet";
const _FG = "FilterGroups";
const _FGi = "FilterGroup";
const _FPR = "FleetProxyRule";
const _FPRl = "FleetProxyRules";
const _FS = "FleetStatus";
const _Fl = "Fleets";
const _GRGT = "GetReportGroupTrend";
const _GRGTI = "GetReportGroupTrendInput";
const _GRGTO = "GetReportGroupTrendOutput";
const _GRP = "GetResourcePolicy";
const _GRPI = "GetResourcePolicyInput";
const _GRPO = "GetResourcePolicyOutput";
const _GSC = "GitSubmodulesConfig";
const _IIE = "InvalidInputException";
const _IPC = "InvalidateProjectCache";
const _IPCI = "InvalidateProjectCacheInput";
const _IPCO = "InvalidateProjectCacheOutput";
const _ISC = "ImportSourceCredentials";
const _ISCI = "ImportSourceCredentialsInput";
const _ISCO = "ImportSourceCredentialsOutput";
const _LB = "ListBuilds";
const _LBB = "ListBuildBatches";
const _LBBFP = "ListBuildBatchesForProject";
const _LBBFPI = "ListBuildBatchesForProjectInput";
const _LBBFPO = "ListBuildBatchesForProjectOutput";
const _LBBI = "ListBuildBatchesInput";
const _LBBO = "ListBuildBatchesOutput";
const _LBFP = "ListBuildsForProject";
const _LBFPI = "ListBuildsForProjectInput";
const _LBFPO = "ListBuildsForProjectOutput";
const _LBI = "ListBuildsInput";
const _LBO = "ListBuildsOutput";
const _LC = "LogsConfig";
const _LCEFS = "ListCommandExecutionsForSandbox";
const _LCEFSI = "ListCommandExecutionsForSandboxInput";
const _LCEFSO = "ListCommandExecutionsForSandboxOutput";
const _LCEI = "ListCuratedEnvironmentImages";
const _LCEII = "ListCuratedEnvironmentImagesInput";
const _LCEIO = "ListCuratedEnvironmentImagesOutput";
const _LF = "ListFleets";
const _LFI = "ListFleetsInput";
const _LFO = "ListFleetsOutput";
const _LL = "LogsLocation";
const _LP = "ListProjects";
const _LPI = "ListProjectsInput";
const _LPO = "ListProjectsOutput";
const _LR = "ListReports";
const _LRFRG = "ListReportsForReportGroup";
const _LRFRGI = "ListReportsForReportGroupInput";
const _LRFRGO = "ListReportsForReportGroupOutput";
const _LRG = "ListReportGroups";
const _LRGI = "ListReportGroupsInput";
const _LRGO = "ListReportGroupsOutput";
const _LRI = "ListReportsInput";
const _LRO = "ListReportsOutput";
const _LS = "ListSandboxes";
const _LSC = "ListSourceCredentials";
const _LSCI = "ListSourceCredentialsInput";
const _LSCO = "ListSourceCredentialsOutput";
const _LSFP = "ListSandboxesForProject";
const _LSFPI = "ListSandboxesForProjectInput";
const _LSFPO = "ListSandboxesForProjectOutput";
const _LSI = "ListSandboxesInput";
const _LSO = "ListSandboxesOutput";
const _LSP = "ListSharedProjects";
const _LSPI = "ListSharedProjectsInput";
const _LSPO = "ListSharedProjectsOutput";
const _LSRG = "ListSharedReportGroups";
const _LSRGI = "ListSharedReportGroupsInput";
const _LSRGO = "ListSharedReportGroupsOutput";
const _NI = "NetworkInterface";
const _OAPE = "OAuthProviderException";
const _P = "Project";
const _PA = "ProjectArtifacts";
const _PAL = "ProjectArtifactsList";
const _PB = "ProjectBadge";
const _PBBC = "ProjectBuildBatchConfig";
const _PC = "PhaseContext";
const _PCh = "PhaseContexts";
const _PCr = "ProjectCache";
const _PCro = "ProxyConfiguration";
const _PE = "ProjectEnvironment";
const _PF = "ProjectFleet";
const _PFSL = "ProjectFileSystemLocation";
const _PFSLr = "ProjectFileSystemLocations";
const _PRBP = "PullRequestBuildPolicy";
const _PRP = "PutResourcePolicy";
const _PRPI = "PutResourcePolicyInput";
const _PRPO = "PutResourcePolicyOutput";
const _PS = "ProjectSource";
const _PSSV = "ProjectSecondarySourceVersions";
const _PSV = "ProjectSourceVersion";
const _PSr = "ProjectSources";
const _Pr = "Projects";
const _R = "Report";
const _RA = "ResolvedArtifact";
const _RAEE = "ResourceAlreadyExistsException";
const _RB = "RetryBuild";
const _RBB = "RetryBuildBatch";
const _RBBI = "RetryBuildBatchInput";
const _RBBO = "RetryBuildBatchOutput";
const _RBI = "RetryBuildInput";
const _RBO = "RetryBuildOutput";
const _RC = "RegistryCredential";
const _REC = "ReportExportConfig";
const _RF = "ReportFilter";
const _RG = "ReportGroup";
const _RGTRDL = "ReportGroupTrendRawDataList";
const _RGTS = "ReportGroupTrendStats";
const _RGe = "ReportGroups";
const _RNFE = "ResourceNotFoundException";
const _RSA = "ResolvedSecondaryArtifacts";
const _RWRD = "ReportWithRawData";
const _Re = "Reports";
const _S = "Sandbox";
const _SA = "SourceAuth";
const _SB = "StartBuild";
const _SBB = "StartBuildBatch";
const _SBBI = "StartBuildBatchInput";
const _SBBIt = "StopBuildBatchInput";
const _SBBO = "StartBuildBatchOutput";
const _SBBOt = "StopBuildBatchOutput";
const _SBBt = "StopBuildBatch";
const _SBI = "StartBuildInput";
const _SBIt = "StopBuildInput";
const _SBO = "StartBuildOutput";
const _SBOt = "StopBuildOutput";
const _SBt = "StopBuild";
const _SC = "ScopeConfiguration";
const _SCE = "StartCommandExecution";
const _SCEI = "StartCommandExecutionInput";
const _SCEO = "StartCommandExecutionOutput";
const _SCI = "ScalingConfigurationInput";
const _SCIo = "SourceCredentialsInfo";
const _SCIou = "SourceCredentialsInfos";
const _SCO = "ScalingConfigurationOutput";
const _SLC = "S3LogsConfig";
const _SNES = "SensitiveNonEmptyString";
const _SREC = "S3ReportExportConfig";
const _SS = "SensitiveString";
const _SSC = "StartSandboxConnection";
const _SSCI = "StartSandboxConnectionInput";
const _SSCO = "StartSandboxConnectionOutput";
const _SSI = "StartSandboxInput";
const _SSIt = "StopSandboxInput";
const _SSMS = "SSMSession";
const _SSO = "StartSandboxOutput";
const _SSOt = "StopSandboxOutput";
const _SSP = "SandboxSessionPhase";
const _SSPa = "SandboxSessionPhases";
const _SSa = "SandboxSession";
const _SSt = "StartSandbox";
const _SSto = "StopSandbox";
const _Sa = "Sandboxes";
const _T = "Tag";
const _TC = "TestCase";
const _TCF = "TestCaseFilter";
const _TCe = "TestCases";
const _TL = "TagList";
const _TRS = "TestReportSummary";
const _TTSC = "TargetTrackingScalingConfiguration";
const _TTSCa = "TargetTrackingScalingConfigurations";
const _UF = "UpdateFleet";
const _UFI = "UpdateFleetInput";
const _UFO = "UpdateFleetOutput";
const _UP = "UpdateProject";
const _UPI = "UpdateProjectInput";
const _UPO = "UpdateProjectOutput";
const _UPV = "UpdateProjectVisibility";
const _UPVI = "UpdateProjectVisibilityInput";
const _UPVO = "UpdateProjectVisibilityOutput";
const _URG = "UpdateReportGroup";
const _URGI = "UpdateReportGroupInput";
const _URGO = "UpdateReportGroupOutput";
const _UW = "UpdateWebhook";
const _UWI = "UpdateWebhookInput";
const _UWO = "UpdateWebhookOutput";
const _VC = "VpcConfig";
const _W = "Webhook";
const _WF = "WebhookFilter";
const _a = "arn";
const _aI = "artifactIdentifier";
const _aO = "artifactsOverride";
const _aR = "approverRoles";
const _aRC = "autoRetryConfig";
const _aRL = "autoRetryLimit";
const _aRLO = "autoRetryLimitOverride";
const _aRN = "autoRetryNumber";
const _aT = "authType";
const _ar = "artifacts";
const _au = "auth";
const _av = "average";
const _b = "builds";
const _bB = "buildBatches";
const _bBA = "buildBatchArn";
const _bBC = "buildBatchConfig";
const _bBCO = "buildBatchConfigOverride";
const _bBN = "buildBatchNumber";
const _bBNF = "buildBatchesNotFound";
const _bBS = "buildBatchStatus";
const _bBu = "buildBatch";
const _bC = "buildComplete";
const _bCP = "branchCoveragePercentage";
const _bCa = "baseCapacity";
const _bCr = "branchesCovered";
const _bD = "buildsDeleted";
const _bE = "badgeEnabled";
const _bF = "branchFilter";
const _bG = "buildGroups";
const _bM = "branchesMissed";
const _bN = "buildNumber";
const _bND = "buildsNotDeleted";
const _bNF = "buildsNotFound";
const _bO = "bucketOwner";
const _bOA = "bucketOwnerAccess";
const _bOu = "buildspecOverride";
const _bRM = "batchReportMode";
const _bRU = "badgeRequestUrl";
const _bS = "buildStatus";
const _bSC = "buildStatusConfig";
const _bSCO = "buildStatusConfigOverride";
const _bT = "buildType";
const _bTIM = "buildTimeoutInMinutes";
const _bTIMO = "buildTimeoutInMinutesOverride";
const _ba = "badge";
const _bu = "buildspec";
const _buc = "bucket";
const _bui = "build";
const _c = "client";
const _cA = "combineArtifacts";
const _cBL = "concurrentBuildLimit";
const _cBS = "currentBuildSummary";
const _cC = "computeConfiguration";
const _cCS = "codeCoverageSummary";
const _cCo = "codeCoverages";
const _cE = "commandExecutions";
const _cEI = "commandExecutionIds";
const _cENF = "commandExecutionsNotFound";
const _cEo = "commandExecution";
const _cN = "cacheNamespace";
const _cO = "certificateOverride";
const _cOa = "cacheOverride";
const _cP = "currentPhase";
const _cPr = "credentialProvider";
const _cS = "currentSession";
const _cT = "computeType";
const _cTA = "computeTypesAllowed";
const _cTO = "computeTypeOverride";
const _cWL = "cloudWatchLogs";
const _cWLA = "cloudWatchLogsArn";
const _ca = "cache";
const _ce = "certificate";
const _co = "complete";
const _com = "command";
const _con = "contexts";
const _cont = "context";
const _cr = "created";
const _cre = "credential";
const _d = "disk";
const _dB = "defaultBehavior";
const _dC = "desiredCapacity";
const _dINS = "durationInNanoSeconds";
const _dIS = "durationInSeconds";
const _dL = "deepLink";
const _dO = "dependsOn";
const _dR = "deleteReports";
const _dS = "debugSession";
const _dSE = "debugSessionEnabled";
const _dSo = "dockerServer";
const _da = "data";
const _de = "description";
const _do = "domain";
const _e = "error";
const _eC = "exitCode";
const _eCT = "exportConfigType";
const _eCx = "exportConfig";
const _eD = "encryptionDisabled";
const _eEV = "exportedEnvironmentVariables";
const _eI = "executionId";
const _eK = "encryptionKey";
const _eKO = "encryptionKeyOverride";
const _eMP = "excludeMatchedPattern";
const _eT = "endTime";
const _eTO = "environmentTypeOverride";
const _eTn = "environmentType";
const _eV = "environmentVariables";
const _eVO = "environmentVariablesOverride";
const _ef = "effect";
const _en = "environment";
const _ent = "entities";
const _ex = "expired";
const _f = "fleets";
const _fA = "fleetsAllowed";
const _fAl = "fleetArn";
const _fG = "filterGroups";
const _fNF = "fleetsNotFound";
const _fO = "fleetOverride";
const _fP = "filePath";
const _fS = "fetchSubmodules";
const _fSL = "fileSystemLocations";
const _fSR = "fleetServiceRole";
const _fi = "filter";
const _fl = "fleet";
const _gCD = "gitCloneDepth";
const _gCDO = "gitCloneDepthOverride";
const _gN = "groupName";
const _gSC = "gitSubmodulesConfig";
const _gSCO = "gitSubmodulesConfigOverride";
const _i = "ids";
const _iF = "ignoreFailure";
const _iI = "imageId";
const _iO = "imageOverride";
const _iPCT = "imagePullCredentialsType";
const _iPCTO = "imagePullCredentialsTypeOverride";
const _iS = "insecureSsl";
const _iSO = "insecureSslOverride";
const _iT = "instanceType";
const _iTd = "idempotencyToken";
const _id = "id";
const _ide = "identifier";
const _im = "images";
const _ima = "image";
const _in = "initiator";
const _k = "key";
const _ke = "keyword";
const _l = "logs";
const _lC = "logConfig";
const _lCO = "logsConfigOverride";
const _lCP = "lineCoveragePercentage";
const _lCi = "linesCovered";
const _lCo = "logsConfig";
const _lM = "linesMissed";
const _lMS = "lastModifiedSecret";
const _lMa = "lastModified";
const _la = "language";
const _lan = "languages";
const _lo = "location";
const _m = "message";
const _mBA = "maximumBuildsAllowed";
const _mC = "manualCreation";
const _mCa = "maxCapacity";
const _mLCP = "minLineCoveragePercentage";
const _mLCPa = "maxLineCoveragePercentage";
const _mO = "mountOptions";
const _mP = "mountPoint";
const _mR = "maxResults";
const _mT = "machineType";
const _mTe = "metricType";
const _ma = "max";
const _md = "md5sum";
const _me = "memory";
const _mi = "min";
const _mo = "modes";
const _n = "names";
const _nAR = "nextAutoRetry";
const _nI = "networkInterface";
const _nII = "networkInterfaceId";
const _nOR = "numOfReports";
const _nT = "nextToken";
const _nTa = "namespaceType";
const _na = "name";
const _oAN = "overrideArtifactName";
const _oB = "overflowBehavior";
const _oPR = "orderedProxyRules";
const _p = "projects";
const _pA = "primaryArtifact";
const _pAR = "previousAutoRetry";
const _pAr = "projectArn";
const _pBSL = "priorBuildSummaryList";
const _pC = "proxyConfiguration";
const _pM = "privilegedMode";
const _pMO = "privilegedModeOverride";
const _pN = "projectName";
const _pNF = "projectsNotFound";
const _pPA = "publicProjectAlias";
const _pRBP = "pullRequestBuildPolicy";
const _pS = "phaseStatus";
const _pT = "phaseType";
const _pU = "payloadUrl";
const _pV = "projectVisibility";
const _pa = "path";
const _pac = "packaging";
const _pat = "pattern";
const _ph = "phases";
const _pl = "platform";
const _pla = "platforms";
const _po = "policy";
const _pr = "project";
const _pre = "prefix";
const _qTIM = "queuedTimeoutInMinutes";
const _qTIMO = "queuedTimeoutInMinutesOverride";
const _r = "reports";
const _rA = "reportArns";
const _rAR = "resourceAccessRole";
const _rARN = "reportARN";
const _rAe = "resourceArn";
const _rAep = "reportArn";
const _rBBSO = "reportBuildBatchStatusOverride";
const _rBS = "reportBuildStatus";
const _rBSO = "reportBuildStatusOverride";
const _rC = "registryCredential";
const _rCA = "requiresCommentApproval";
const _rCO = "registryCredentialOverride";
const _rD = "rawData";
const _rG = "reportGroups";
const _rGA = "reportGroupArns";
const _rGAe = "reportGroupArn";
const _rGNF = "reportGroupsNotFound";
const _rGe = "reportGroup";
const _rNF = "reportsNotFound";
const _rO = "requestedOn";
const _rS = "rotateSecret";
const _rSV = "resolvedSourceVersion";
const _rT = "retryType";
const _rTe = "requestTime";
const _re = "restrictions";
const _res = "resource";
const _s = "sandboxes";
const _sA = "secondaryArtifacts";
const _sAO = "secondaryArtifactsOverride";
const _sAOo = "sourceAuthOverride";
const _sAa = "sandboxArn";
const _sB = "sortBy";
const _sC = "statusCode";
const _sCI = "sourceCredentialsInfos";
const _sCc = "scalingConfiguration";
const _sCco = "scopeConfiguration";
const _sCt = "statusCounts";
const _sD = "s3Destination";
const _sDL = "s3DeepLink";
const _sE = "sessionEnabled";
const _sEC = "standardErrContent";
const _sGI = "securityGroupIds";
const _sI = "sandboxId";
const _sIe = "sessionId";
const _sIo = "sourceIdentifier";
const _sIu = "subnetId";
const _sL = "s3Logs";
const _sLA = "s3LogsArn";
const _sLO = "sourceLocationOverride";
const _sM = "statusMessage";
const _sN = "streamName";
const _sNF = "sandboxesNotFound";
const _sO = "sortOrder";
const _sOC = "standardOutputContent";
const _sOh = "shouldOverwrite";
const _sR = "serviceRole";
const _sRO = "serviceRoleOverride";
const _sS = "secondarySources";
const _sSO = "secondarySourcesOverride";
const _sSV = "secondarySourceVersions";
const _sSVO = "secondarySourcesVersionOverride";
const _sSs = "ssmSession";
const _sT = "startTime";
const _sTO = "sourceTypeOverride";
const _sTc = "scalingType";
const _sTe = "sessionTarget";
const _sTer = "serverType";
const _sTu = "submitTime";
const _sU = "streamUrl";
const _sV = "sourceVersion";
const _sa = "sandbox";
const _sc = "scope";
const _se = "secret";
const _sh = "sha256sum";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.codebuild";
const _so = "source";
const _st = "status";
const _sta = "stats";
const _su = "subnets";
const _t = "type";
const _tC = "testCases";
const _tF = "trendField";
const _tIM = "timeoutInMinutes";
const _tIMO = "timeoutInMinutesOverride";
const _tIMi = "timeoutInMins";
const _tRDP = "testRawDataPath";
const _tS = "testSummary";
const _tSN = "testSuiteName";
const _tTSC = "targetTrackingScalingConfigs";
const _tU = "targetUrl";
const _tV = "tokenValue";
const _tVa = "targetValue";
const _ta = "tags";
const _to = "token";
const _tot = "total";
const _tr = "truncated";
const _u = "username";
const _ur = "url";
const _v = "versions";
const _vC = "vpcConfig";
const _vCp = "vCpu";
const _vI = "vpcId";
const _va = "value";
const _w = "webhook";
const n0 = "com.amazonaws.codebuild";
var SensitiveNonEmptyString = [0, n0, _SNES, 8, 0];
var SensitiveString = [0, n0, _SS, 8, 0];
var AccountLimitExceededException$ = [-3, n0, _ALEE,
    { [_e]: _c },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(AccountLimitExceededException$, AccountLimitExceededException);
var AccountSuspendedException$ = [-3, n0, _ASE,
    { [_e]: _c },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(AccountSuspendedException$, AccountSuspendedException);
var AutoRetryConfig$ = [3, n0, _ARC,
    0,
    [_aRL, _aRN, _nAR, _pAR],
    [1, 1, 0, 0]
];
var BatchDeleteBuildsInput$ = [3, n0, _BDBI,
    0,
    [_i],
    [64 | 0], 1
];
var BatchDeleteBuildsOutput$ = [3, n0, _BDBO,
    0,
    [_bD, _bND],
    [64 | 0, () => BuildsNotDeleted]
];
var BatchGetBuildBatchesInput$ = [3, n0, _BGBBI,
    0,
    [_i],
    [64 | 0], 1
];
var BatchGetBuildBatchesOutput$ = [3, n0, _BGBBO,
    0,
    [_bB, _bBNF],
    [() => BuildBatches, 64 | 0]
];
var BatchGetBuildsInput$ = [3, n0, _BGBI,
    0,
    [_i],
    [64 | 0], 1
];
var BatchGetBuildsOutput$ = [3, n0, _BGBO,
    0,
    [_b, _bNF],
    [() => Builds, 64 | 0]
];
var BatchGetCommandExecutionsInput$ = [3, n0, _BGCEI,
    0,
    [_sI, _cEI],
    [0, 64 | 0], 2
];
var BatchGetCommandExecutionsOutput$ = [3, n0, _BGCEO,
    0,
    [_cE, _cENF],
    [[() => CommandExecutions, 0], 64 | 0]
];
var BatchGetFleetsInput$ = [3, n0, _BGFI,
    0,
    [_n],
    [64 | 0], 1
];
var BatchGetFleetsOutput$ = [3, n0, _BGFO,
    0,
    [_f, _fNF],
    [() => Fleets, 64 | 0]
];
var BatchGetProjectsInput$ = [3, n0, _BGPI,
    0,
    [_n],
    [64 | 0], 1
];
var BatchGetProjectsOutput$ = [3, n0, _BGPO,
    0,
    [_p, _pNF],
    [() => Projects, 64 | 0]
];
var BatchGetReportGroupsInput$ = [3, n0, _BGRGI,
    0,
    [_rGA],
    [64 | 0], 1
];
var BatchGetReportGroupsOutput$ = [3, n0, _BGRGO,
    0,
    [_rG, _rGNF],
    [() => ReportGroups, 64 | 0]
];
var BatchGetReportsInput$ = [3, n0, _BGRI,
    0,
    [_rA],
    [64 | 0], 1
];
var BatchGetReportsOutput$ = [3, n0, _BGRO,
    0,
    [_r, _rNF],
    [() => Reports, 64 | 0]
];
var BatchGetSandboxesInput$ = [3, n0, _BGSI,
    0,
    [_i],
    [64 | 0], 1
];
var BatchGetSandboxesOutput$ = [3, n0, _BGSO,
    0,
    [_s, _sNF],
    [() => Sandboxes, 64 | 0]
];
var BatchRestrictions$ = [3, n0, _BR,
    0,
    [_mBA, _cTA, _fA],
    [1, 64 | 0, 64 | 0]
];
var Build$ = [3, n0, _B,
    0,
    [_id, _a, _bN, _sT, _eT, _cP, _bS, _sV, _rSV, _pN, _ph, _so, _sS, _sSV, _ar, _sA, _ca, _en, _sR, _l, _tIM, _qTIM, _bC, _in, _vC, _nI, _eK, _eEV, _rA, _fSL, _dS, _bBA, _aRC],
    [0, 0, 1, 4, 4, 0, 0, 0, 0, 0, () => BuildPhases, () => ProjectSource$, () => ProjectSources, () => ProjectSecondarySourceVersions, () => BuildArtifacts$, () => BuildArtifactsList, () => ProjectCache$, () => ProjectEnvironment$, 0, () => LogsLocation$, 1, 1, 2, 0, () => VpcConfig$, () => NetworkInterface$, 0, () => ExportedEnvironmentVariables, 64 | 0, () => ProjectFileSystemLocations, () => DebugSession$, 0, () => AutoRetryConfig$]
];
var BuildArtifacts$ = [3, n0, _BA,
    0,
    [_lo, _sh, _md, _oAN, _eD, _aI, _bOA],
    [0, 0, 0, 2, 2, 0, 0]
];
var BuildBatch$ = [3, n0, _BB,
    0,
    [_id, _a, _sT, _eT, _cP, _bBS, _sV, _rSV, _pN, _ph, _so, _sS, _sSV, _ar, _sA, _ca, _en, _sR, _lC, _bTIM, _qTIM, _co, _in, _vC, _eK, _bBN, _fSL, _bBC, _bG, _dSE, _rA],
    [0, 0, 4, 4, 0, 0, 0, 0, 0, () => BuildBatchPhases, () => ProjectSource$, () => ProjectSources, () => ProjectSecondarySourceVersions, () => BuildArtifacts$, () => BuildArtifactsList, () => ProjectCache$, () => ProjectEnvironment$, 0, () => LogsConfig$, 1, 1, 2, 0, () => VpcConfig$, 0, 1, () => ProjectFileSystemLocations, () => ProjectBuildBatchConfig$, () => BuildGroups, 2, 64 | 0]
];
var BuildBatchFilter$ = [3, n0, _BBF,
    0,
    [_st],
    [0]
];
var BuildBatchPhase$ = [3, n0, _BBP,
    0,
    [_pT, _pS, _sT, _eT, _dIS, _con],
    [0, 0, 4, 4, 1, () => PhaseContexts]
];
var BuildGroup$ = [3, n0, _BG,
    0,
    [_ide, _dO, _iF, _cBS, _pBSL],
    [0, 64 | 0, 2, () => BuildSummary$, () => BuildSummaries]
];
var BuildNotDeleted$ = [3, n0, _BND,
    0,
    [_id, _sC],
    [0, 0]
];
var BuildPhase$ = [3, n0, _BP,
    0,
    [_pT, _pS, _sT, _eT, _dIS, _con],
    [0, 0, 4, 4, 1, () => PhaseContexts]
];
var BuildStatusConfig$ = [3, n0, _BSC,
    0,
    [_cont, _tU],
    [0, 0]
];
var BuildSummary$ = [3, n0, _BS,
    0,
    [_a, _rO, _bS, _pA, _sA],
    [0, 4, 0, () => ResolvedArtifact$, () => ResolvedSecondaryArtifacts]
];
var CloudWatchLogsConfig$ = [3, n0, _CWLC,
    0,
    [_st, _gN, _sN],
    [0, 0, 0], 1
];
var CodeCoverage$ = [3, n0, _CC,
    0,
    [_id, _rARN, _fP, _lCP, _lCi, _lM, _bCP, _bCr, _bM, _ex],
    [0, 0, 0, 1, 1, 1, 1, 1, 1, 4]
];
var CodeCoverageReportSummary$ = [3, n0, _CCRS,
    0,
    [_lCP, _lCi, _lM, _bCP, _bCr, _bM],
    [1, 1, 1, 1, 1, 1]
];
var CommandExecution$ = [3, n0, _CE,
    0,
    [_id, _sI, _sTu, _sT, _eT, _st, _com, _t, _eC, _sOC, _sEC, _l, _sAa],
    [0, 0, 4, 4, 4, 0, [() => SensitiveNonEmptyString, 0], 0, 0, [() => SensitiveNonEmptyString, 0], [() => SensitiveNonEmptyString, 0], () => LogsLocation$, 0]
];
var ComputeConfiguration$ = [3, n0, _CCo,
    0,
    [_vCp, _me, _d, _mT, _iT],
    [1, 1, 1, 0, 0]
];
var CreateFleetInput$ = [3, n0, _CFI,
    0,
    [_na, _bCa, _eTn, _cT, _cC, _sCc, _oB, _vC, _pC, _iI, _fSR, _ta],
    [0, 1, 0, 0, () => ComputeConfiguration$, () => ScalingConfigurationInput$, 0, () => VpcConfig$, () => ProxyConfiguration$, 0, 0, () => TagList], 4
];
var CreateFleetOutput$ = [3, n0, _CFO,
    0,
    [_fl],
    [() => Fleet$]
];
var CreateProjectInput$ = [3, n0, _CPI,
    0,
    [_na, _so, _ar, _en, _sR, _de, _sS, _sV, _sSV, _sA, _ca, _tIM, _qTIM, _eK, _ta, _vC, _bE, _lCo, _fSL, _bBC, _cBL, _aRL],
    [0, () => ProjectSource$, () => ProjectArtifacts$, () => ProjectEnvironment$, 0, 0, () => ProjectSources, 0, () => ProjectSecondarySourceVersions, () => ProjectArtifactsList, () => ProjectCache$, 1, 1, 0, () => TagList, () => VpcConfig$, 2, () => LogsConfig$, () => ProjectFileSystemLocations, () => ProjectBuildBatchConfig$, 1, 1], 5
];
var CreateProjectOutput$ = [3, n0, _CPO,
    0,
    [_pr],
    [() => Project$]
];
var CreateReportGroupInput$ = [3, n0, _CRGI,
    0,
    [_na, _t, _eCx, _ta],
    [0, 0, () => ReportExportConfig$, () => TagList], 3
];
var CreateReportGroupOutput$ = [3, n0, _CRGO,
    0,
    [_rGe],
    [() => ReportGroup$]
];
var CreateWebhookInput$ = [3, n0, _CWI,
    0,
    [_pN, _bF, _fG, _bT, _mC, _sCco, _pRBP],
    [0, 0, () => FilterGroups, 0, 2, () => ScopeConfiguration$, () => PullRequestBuildPolicy$], 1
];
var CreateWebhookOutput$ = [3, n0, _CWO,
    0,
    [_w],
    [() => Webhook$]
];
var DebugSession$ = [3, n0, _DS,
    0,
    [_sE, _sTe],
    [2, 0]
];
var DeleteBuildBatchInput$ = [3, n0, _DBBI,
    0,
    [_id],
    [0], 1
];
var DeleteBuildBatchOutput$ = [3, n0, _DBBO,
    0,
    [_sC, _bD, _bND],
    [0, 64 | 0, () => BuildsNotDeleted]
];
var DeleteFleetInput$ = [3, n0, _DFI,
    0,
    [_a],
    [0], 1
];
var DeleteFleetOutput$ = [3, n0, _DFO,
    0,
    [],
    []
];
var DeleteProjectInput$ = [3, n0, _DPI,
    0,
    [_na],
    [0], 1
];
var DeleteProjectOutput$ = [3, n0, _DPO,
    0,
    [],
    []
];
var DeleteReportGroupInput$ = [3, n0, _DRGI,
    0,
    [_a, _dR],
    [0, 2], 1
];
var DeleteReportGroupOutput$ = [3, n0, _DRGO,
    0,
    [],
    []
];
var DeleteReportInput$ = [3, n0, _DRI,
    0,
    [_a],
    [0], 1
];
var DeleteReportOutput$ = [3, n0, _DRO,
    0,
    [],
    []
];
var DeleteResourcePolicyInput$ = [3, n0, _DRPI,
    0,
    [_rAe],
    [0], 1
];
var DeleteResourcePolicyOutput$ = [3, n0, _DRPO,
    0,
    [],
    []
];
var DeleteSourceCredentialsInput$ = [3, n0, _DSCI,
    0,
    [_a],
    [0], 1
];
var DeleteSourceCredentialsOutput$ = [3, n0, _DSCO,
    0,
    [_a],
    [0]
];
var DeleteWebhookInput$ = [3, n0, _DWI,
    0,
    [_pN],
    [0], 1
];
var DeleteWebhookOutput$ = [3, n0, _DWO,
    0,
    [],
    []
];
var DescribeCodeCoveragesInput$ = [3, n0, _DCCI,
    0,
    [_rAep, _nT, _mR, _sO, _sB, _mLCP, _mLCPa],
    [0, 0, 1, 0, 0, 1, 1], 1
];
var DescribeCodeCoveragesOutput$ = [3, n0, _DCCO,
    0,
    [_nT, _cCo],
    [0, () => CodeCoverages]
];
var DescribeTestCasesInput$ = [3, n0, _DTCI,
    0,
    [_rAep, _nT, _mR, _fi],
    [0, 0, 1, () => TestCaseFilter$], 1
];
var DescribeTestCasesOutput$ = [3, n0, _DTCO,
    0,
    [_nT, _tC],
    [0, () => TestCases]
];
var DockerServer$ = [3, n0, _DSo,
    0,
    [_cT, _sGI, _st],
    [0, 64 | 0, () => DockerServerStatus$], 1
];
var DockerServerStatus$ = [3, n0, _DSS,
    0,
    [_st, _m],
    [0, 0]
];
var EnvironmentImage$ = [3, n0, _EI,
    0,
    [_na, _de, _v],
    [0, 0, 64 | 0]
];
var EnvironmentLanguage$ = [3, n0, _EL,
    0,
    [_la, _im],
    [0, () => EnvironmentImages]
];
var EnvironmentPlatform$ = [3, n0, _EP,
    0,
    [_pl, _lan],
    [0, () => EnvironmentLanguages]
];
var EnvironmentVariable$ = [3, n0, _EV,
    0,
    [_na, _va, _t],
    [0, 0, 0], 2
];
var ExportedEnvironmentVariable$ = [3, n0, _EEV,
    0,
    [_na, _va],
    [0, 0]
];
var Fleet$ = [3, n0, _F,
    0,
    [_a, _na, _id, _cr, _lMa, _st, _bCa, _eTn, _cT, _cC, _sCc, _oB, _vC, _pC, _iI, _fSR, _ta],
    [0, 0, 0, 4, 4, () => FleetStatus$, 1, 0, 0, () => ComputeConfiguration$, () => ScalingConfigurationOutput$, 0, () => VpcConfig$, () => ProxyConfiguration$, 0, 0, () => TagList]
];
var FleetProxyRule$ = [3, n0, _FPR,
    0,
    [_t, _ef, _ent],
    [0, 0, 64 | 0], 3
];
var FleetStatus$ = [3, n0, _FS,
    0,
    [_sC, _cont, _m],
    [0, 0, 0]
];
var GetReportGroupTrendInput$ = [3, n0, _GRGTI,
    0,
    [_rGAe, _tF, _nOR],
    [0, 0, 1], 2
];
var GetReportGroupTrendOutput$ = [3, n0, _GRGTO,
    0,
    [_sta, _rD],
    [() => ReportGroupTrendStats$, () => ReportGroupTrendRawDataList]
];
var GetResourcePolicyInput$ = [3, n0, _GRPI,
    0,
    [_rAe],
    [0], 1
];
var GetResourcePolicyOutput$ = [3, n0, _GRPO,
    0,
    [_po],
    [0]
];
var GitSubmodulesConfig$ = [3, n0, _GSC,
    0,
    [_fS],
    [2], 1
];
var ImportSourceCredentialsInput$ = [3, n0, _ISCI,
    0,
    [_to, _sTer, _aT, _u, _sOh],
    [[() => SensitiveNonEmptyString, 0], 0, 0, 0, 2], 3
];
var ImportSourceCredentialsOutput$ = [3, n0, _ISCO,
    0,
    [_a],
    [0]
];
var InvalidateProjectCacheInput$ = [3, n0, _IPCI,
    0,
    [_pN],
    [0], 1
];
var InvalidateProjectCacheOutput$ = [3, n0, _IPCO,
    0,
    [],
    []
];
var InvalidInputException$ = [-3, n0, _IIE,
    { [_e]: _c },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InvalidInputException$, InvalidInputException);
var ListBuildBatchesForProjectInput$ = [3, n0, _LBBFPI,
    0,
    [_pN, _fi, _mR, _sO, _nT],
    [0, () => BuildBatchFilter$, 1, 0, 0]
];
var ListBuildBatchesForProjectOutput$ = [3, n0, _LBBFPO,
    0,
    [_i, _nT],
    [64 | 0, 0]
];
var ListBuildBatchesInput$ = [3, n0, _LBBI,
    0,
    [_fi, _mR, _sO, _nT],
    [() => BuildBatchFilter$, 1, 0, 0]
];
var ListBuildBatchesOutput$ = [3, n0, _LBBO,
    0,
    [_i, _nT],
    [64 | 0, 0]
];
var ListBuildsForProjectInput$ = [3, n0, _LBFPI,
    0,
    [_pN, _sO, _nT],
    [0, 0, 0], 1
];
var ListBuildsForProjectOutput$ = [3, n0, _LBFPO,
    0,
    [_i, _nT],
    [64 | 0, 0]
];
var ListBuildsInput$ = [3, n0, _LBI,
    0,
    [_sO, _nT],
    [0, 0]
];
var ListBuildsOutput$ = [3, n0, _LBO,
    0,
    [_i, _nT],
    [64 | 0, 0]
];
var ListCommandExecutionsForSandboxInput$ = [3, n0, _LCEFSI,
    0,
    [_sI, _mR, _sO, _nT],
    [0, 1, 0, [() => SensitiveString, 0]], 1
];
var ListCommandExecutionsForSandboxOutput$ = [3, n0, _LCEFSO,
    0,
    [_cE, _nT],
    [[() => CommandExecutions, 0], 0]
];
var ListCuratedEnvironmentImagesInput$ = [3, n0, _LCEII,
    0,
    [],
    []
];
var ListCuratedEnvironmentImagesOutput$ = [3, n0, _LCEIO,
    0,
    [_pla],
    [() => EnvironmentPlatforms]
];
var ListFleetsInput$ = [3, n0, _LFI,
    0,
    [_nT, _mR, _sO, _sB],
    [[() => SensitiveString, 0], 1, 0, 0]
];
var ListFleetsOutput$ = [3, n0, _LFO,
    0,
    [_nT, _f],
    [0, 64 | 0]
];
var ListProjectsInput$ = [3, n0, _LPI,
    0,
    [_sB, _sO, _nT],
    [0, 0, 0]
];
var ListProjectsOutput$ = [3, n0, _LPO,
    0,
    [_nT, _p],
    [0, 64 | 0]
];
var ListReportGroupsInput$ = [3, n0, _LRGI,
    0,
    [_sO, _sB, _nT, _mR],
    [0, 0, 0, 1]
];
var ListReportGroupsOutput$ = [3, n0, _LRGO,
    0,
    [_nT, _rG],
    [0, 64 | 0]
];
var ListReportsForReportGroupInput$ = [3, n0, _LRFRGI,
    0,
    [_rGAe, _nT, _sO, _mR, _fi],
    [0, 0, 0, 1, () => ReportFilter$], 1
];
var ListReportsForReportGroupOutput$ = [3, n0, _LRFRGO,
    0,
    [_nT, _r],
    [0, 64 | 0]
];
var ListReportsInput$ = [3, n0, _LRI,
    0,
    [_sO, _nT, _mR, _fi],
    [0, 0, 1, () => ReportFilter$]
];
var ListReportsOutput$ = [3, n0, _LRO,
    0,
    [_nT, _r],
    [0, 64 | 0]
];
var ListSandboxesForProjectInput$ = [3, n0, _LSFPI,
    0,
    [_pN, _mR, _sO, _nT],
    [0, 1, 0, [() => SensitiveString, 0]], 1
];
var ListSandboxesForProjectOutput$ = [3, n0, _LSFPO,
    0,
    [_i, _nT],
    [64 | 0, 0]
];
var ListSandboxesInput$ = [3, n0, _LSI,
    0,
    [_mR, _sO, _nT],
    [1, 0, 0]
];
var ListSandboxesOutput$ = [3, n0, _LSO,
    0,
    [_i, _nT],
    [64 | 0, 0]
];
var ListSharedProjectsInput$ = [3, n0, _LSPI,
    0,
    [_sB, _sO, _mR, _nT],
    [0, 0, 1, 0]
];
var ListSharedProjectsOutput$ = [3, n0, _LSPO,
    0,
    [_nT, _p],
    [0, 64 | 0]
];
var ListSharedReportGroupsInput$ = [3, n0, _LSRGI,
    0,
    [_sO, _sB, _nT, _mR],
    [0, 0, 0, 1]
];
var ListSharedReportGroupsOutput$ = [3, n0, _LSRGO,
    0,
    [_nT, _rG],
    [0, 64 | 0]
];
var ListSourceCredentialsInput$ = [3, n0, _LSCI,
    0,
    [],
    []
];
var ListSourceCredentialsOutput$ = [3, n0, _LSCO,
    0,
    [_sCI],
    [() => SourceCredentialsInfos]
];
var LogsConfig$ = [3, n0, _LC,
    0,
    [_cWL, _sL],
    [() => CloudWatchLogsConfig$, () => S3LogsConfig$]
];
var LogsLocation$ = [3, n0, _LL,
    0,
    [_gN, _sN, _dL, _sDL, _cWLA, _sLA, _cWL, _sL],
    [0, 0, 0, 0, 0, 0, () => CloudWatchLogsConfig$, () => S3LogsConfig$]
];
var NetworkInterface$ = [3, n0, _NI,
    0,
    [_sIu, _nII],
    [0, 0]
];
var OAuthProviderException$ = [-3, n0, _OAPE,
    { [_e]: _c },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(OAuthProviderException$, OAuthProviderException);
var PhaseContext$ = [3, n0, _PC,
    0,
    [_sC, _m],
    [0, 0]
];
var Project$ = [3, n0, _P,
    0,
    [_na, _a, _de, _so, _sS, _sV, _sSV, _ar, _sA, _ca, _en, _sR, _tIM, _qTIM, _eK, _ta, _cr, _lMa, _w, _vC, _ba, _lCo, _fSL, _bBC, _cBL, _pV, _pPA, _rAR, _aRL],
    [0, 0, 0, () => ProjectSource$, () => ProjectSources, 0, () => ProjectSecondarySourceVersions, () => ProjectArtifacts$, () => ProjectArtifactsList, () => ProjectCache$, () => ProjectEnvironment$, 0, 1, 1, 0, () => TagList, 4, 4, () => Webhook$, () => VpcConfig$, () => ProjectBadge$, () => LogsConfig$, () => ProjectFileSystemLocations, () => ProjectBuildBatchConfig$, 1, 0, 0, 0, 1]
];
var ProjectArtifacts$ = [3, n0, _PA,
    0,
    [_t, _lo, _pa, _nTa, _na, _pac, _oAN, _eD, _aI, _bOA],
    [0, 0, 0, 0, 0, 0, 2, 2, 0, 0], 1
];
var ProjectBadge$ = [3, n0, _PB,
    0,
    [_bE, _bRU],
    [2, 0]
];
var ProjectBuildBatchConfig$ = [3, n0, _PBBC,
    0,
    [_sR, _cA, _re, _tIMi, _bRM],
    [0, 2, () => BatchRestrictions$, 1, 0]
];
var ProjectCache$ = [3, n0, _PCr,
    0,
    [_t, _lo, _mo, _cN],
    [0, 0, 64 | 0, 0], 1
];
var ProjectEnvironment$ = [3, n0, _PE,
    0,
    [_t, _ima, _cT, _cC, _fl, _eV, _pM, _ce, _rC, _iPCT, _dSo],
    [0, 0, 0, () => ComputeConfiguration$, () => ProjectFleet$, () => EnvironmentVariables, 2, 0, () => RegistryCredential$, 0, () => DockerServer$], 3
];
var ProjectFileSystemLocation$ = [3, n0, _PFSL,
    0,
    [_t, _lo, _mP, _ide, _mO],
    [0, 0, 0, 0, 0]
];
var ProjectFleet$ = [3, n0, _PF,
    0,
    [_fAl],
    [0]
];
var ProjectSource$ = [3, n0, _PS,
    0,
    [_t, _lo, _gCD, _gSC, _bu, _au, _rBS, _bSC, _iS, _sIo],
    [0, 0, 1, () => GitSubmodulesConfig$, 0, () => SourceAuth$, 2, () => BuildStatusConfig$, 2, 0], 1
];
var ProjectSourceVersion$ = [3, n0, _PSV,
    0,
    [_sIo, _sV],
    [0, 0], 2
];
var ProxyConfiguration$ = [3, n0, _PCro,
    0,
    [_dB, _oPR],
    [0, () => FleetProxyRules]
];
var PullRequestBuildPolicy$ = [3, n0, _PRBP,
    0,
    [_rCA, _aR],
    [0, 64 | 0], 1
];
var PutResourcePolicyInput$ = [3, n0, _PRPI,
    0,
    [_po, _rAe],
    [0, 0], 2
];
var PutResourcePolicyOutput$ = [3, n0, _PRPO,
    0,
    [_rAe],
    [0]
];
var RegistryCredential$ = [3, n0, _RC,
    0,
    [_cre, _cPr],
    [0, 0], 2
];
var Report$ = [3, n0, _R,
    0,
    [_a, _t, _na, _rGAe, _eI, _st, _cr, _ex, _eCx, _tr, _tS, _cCS],
    [0, 0, 0, 0, 0, 0, 4, 4, () => ReportExportConfig$, 2, () => TestReportSummary$, () => CodeCoverageReportSummary$]
];
var ReportExportConfig$ = [3, n0, _REC,
    0,
    [_eCT, _sD],
    [0, () => S3ReportExportConfig$]
];
var ReportFilter$ = [3, n0, _RF,
    0,
    [_st],
    [0]
];
var ReportGroup$ = [3, n0, _RG,
    0,
    [_a, _na, _t, _eCx, _cr, _lMa, _ta, _st],
    [0, 0, 0, () => ReportExportConfig$, 4, 4, () => TagList, 0]
];
var ReportGroupTrendStats$ = [3, n0, _RGTS,
    0,
    [_av, _ma, _mi],
    [0, 0, 0]
];
var ReportWithRawData$ = [3, n0, _RWRD,
    0,
    [_rAep, _da],
    [0, 0]
];
var ResolvedArtifact$ = [3, n0, _RA,
    0,
    [_t, _lo, _ide],
    [0, 0, 0]
];
var ResourceAlreadyExistsException$ = [-3, n0, _RAEE,
    { [_e]: _c },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ResourceAlreadyExistsException$, ResourceAlreadyExistsException);
var ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ResourceNotFoundException$, ResourceNotFoundException);
var RetryBuildBatchInput$ = [3, n0, _RBBI,
    0,
    [_id, _iTd, _rT],
    [0, 0, 0]
];
var RetryBuildBatchOutput$ = [3, n0, _RBBO,
    0,
    [_bBu],
    [() => BuildBatch$]
];
var RetryBuildInput$ = [3, n0, _RBI,
    0,
    [_id, _iTd],
    [0, 0]
];
var RetryBuildOutput$ = [3, n0, _RBO,
    0,
    [_bui],
    [() => Build$]
];
var S3LogsConfig$ = [3, n0, _SLC,
    0,
    [_st, _lo, _eD, _bOA],
    [0, 0, 2, 0], 1
];
var S3ReportExportConfig$ = [3, n0, _SREC,
    0,
    [_buc, _bO, _pa, _pac, _eK, _eD],
    [0, 0, 0, 0, 0, 2]
];
var Sandbox$ = [3, n0, _S,
    0,
    [_id, _a, _pN, _rTe, _sT, _eT, _st, _so, _sV, _sS, _sSV, _en, _fSL, _tIM, _qTIM, _vC, _lC, _eK, _sR, _cS],
    [0, 0, 0, 4, 4, 4, 0, () => ProjectSource$, 0, () => ProjectSources, () => ProjectSecondarySourceVersions, () => ProjectEnvironment$, () => ProjectFileSystemLocations, 1, 1, () => VpcConfig$, () => LogsConfig$, 0, 0, () => SandboxSession$]
];
var SandboxSession$ = [3, n0, _SSa,
    0,
    [_id, _st, _sT, _eT, _cP, _ph, _rSV, _l, _nI],
    [0, 0, 4, 4, 0, () => SandboxSessionPhases, 0, () => LogsLocation$, () => NetworkInterface$]
];
var SandboxSessionPhase$ = [3, n0, _SSP,
    0,
    [_pT, _pS, _sT, _eT, _dIS, _con],
    [0, 0, 4, 4, 1, () => PhaseContexts]
];
var ScalingConfigurationInput$ = [3, n0, _SCI,
    0,
    [_sTc, _tTSC, _mCa],
    [0, () => TargetTrackingScalingConfigurations, 1]
];
var ScalingConfigurationOutput$ = [3, n0, _SCO,
    0,
    [_sTc, _tTSC, _mCa, _dC],
    [0, () => TargetTrackingScalingConfigurations, 1, 1]
];
var ScopeConfiguration$ = [3, n0, _SC,
    0,
    [_na, _sc, _do],
    [0, 0, 0], 2
];
var SourceAuth$ = [3, n0, _SA,
    0,
    [_t, _res],
    [0, 0], 1
];
var SourceCredentialsInfo$ = [3, n0, _SCIo,
    0,
    [_a, _sTer, _aT, _res],
    [0, 0, 0, 0]
];
var SSMSession$ = [3, n0, _SSMS,
    0,
    [_sIe, _tV, _sU],
    [0, 0, 0]
];
var StartBuildBatchInput$ = [3, n0, _SBBI,
    0,
    [_pN, _sSO, _sSVO, _sV, _aO, _sAO, _eVO, _sTO, _sLO, _sAOo, _gCDO, _gSCO, _bOu, _iSO, _rBBSO, _eTO, _iO, _cTO, _cO, _cOa, _sRO, _pMO, _bTIMO, _qTIMO, _eKO, _iTd, _lCO, _rCO, _iPCTO, _bBCO, _dSE],
    [0, () => ProjectSources, () => ProjectSecondarySourceVersions, 0, () => ProjectArtifacts$, () => ProjectArtifactsList, () => EnvironmentVariables, 0, 0, () => SourceAuth$, 1, () => GitSubmodulesConfig$, 0, 2, 2, 0, 0, 0, 0, () => ProjectCache$, 0, 2, 1, 1, 0, 0, () => LogsConfig$, () => RegistryCredential$, 0, () => ProjectBuildBatchConfig$, 2], 1
];
var StartBuildBatchOutput$ = [3, n0, _SBBO,
    0,
    [_bBu],
    [() => BuildBatch$]
];
var StartBuildInput$ = [3, n0, _SBI,
    0,
    [_pN, _sSO, _sSVO, _sV, _aO, _sAO, _eVO, _sTO, _sLO, _sAOo, _gCDO, _gSCO, _bOu, _iSO, _rBSO, _bSCO, _eTO, _iO, _cTO, _cO, _cOa, _sRO, _pMO, _tIMO, _qTIMO, _eKO, _iTd, _lCO, _rCO, _iPCTO, _dSE, _fO, _aRLO],
    [0, () => ProjectSources, () => ProjectSecondarySourceVersions, 0, () => ProjectArtifacts$, () => ProjectArtifactsList, () => EnvironmentVariables, 0, 0, () => SourceAuth$, 1, () => GitSubmodulesConfig$, 0, 2, 2, () => BuildStatusConfig$, 0, 0, 0, 0, () => ProjectCache$, 0, 2, 1, 1, 0, 0, () => LogsConfig$, () => RegistryCredential$, 0, 2, () => ProjectFleet$, 1], 1
];
var StartBuildOutput$ = [3, n0, _SBO,
    0,
    [_bui],
    [() => Build$]
];
var StartCommandExecutionInput$ = [3, n0, _SCEI,
    0,
    [_sI, _com, _t],
    [0, [() => SensitiveNonEmptyString, 0], 0], 2
];
var StartCommandExecutionOutput$ = [3, n0, _SCEO,
    0,
    [_cEo],
    [[() => CommandExecution$, 0]]
];
var StartSandboxConnectionInput$ = [3, n0, _SSCI,
    0,
    [_sI],
    [0], 1
];
var StartSandboxConnectionOutput$ = [3, n0, _SSCO,
    0,
    [_sSs],
    [() => SSMSession$]
];
var StartSandboxInput$ = [3, n0, _SSI,
    0,
    [_pN, _iTd],
    [0, [() => SensitiveString, 0]]
];
var StartSandboxOutput$ = [3, n0, _SSO,
    0,
    [_sa],
    [() => Sandbox$]
];
var StopBuildBatchInput$ = [3, n0, _SBBIt,
    0,
    [_id],
    [0], 1
];
var StopBuildBatchOutput$ = [3, n0, _SBBOt,
    0,
    [_bBu],
    [() => BuildBatch$]
];
var StopBuildInput$ = [3, n0, _SBIt,
    0,
    [_id],
    [0], 1
];
var StopBuildOutput$ = [3, n0, _SBOt,
    0,
    [_bui],
    [() => Build$]
];
var StopSandboxInput$ = [3, n0, _SSIt,
    0,
    [_id],
    [0], 1
];
var StopSandboxOutput$ = [3, n0, _SSOt,
    0,
    [_sa],
    [() => Sandbox$]
];
var Tag$ = [3, n0, _T,
    0,
    [_k, _va],
    [0, 0]
];
var TargetTrackingScalingConfiguration$ = [3, n0, _TTSC,
    0,
    [_mTe, _tVa],
    [0, 1]
];
var TestCase$ = [3, n0, _TC,
    0,
    [_rAep, _tRDP, _pre, _na, _st, _dINS, _m, _ex, _tSN],
    [0, 0, 0, 0, 0, 1, 0, 4, 0]
];
var TestCaseFilter$ = [3, n0, _TCF,
    0,
    [_st, _ke],
    [0, 0]
];
var TestReportSummary$ = [3, n0, _TRS,
    0,
    [_tot, _sCt, _dINS],
    [1, 128 | 1, 1], 3
];
var UpdateFleetInput$ = [3, n0, _UFI,
    0,
    [_a, _bCa, _eTn, _cT, _cC, _sCc, _oB, _vC, _pC, _iI, _fSR, _ta],
    [0, 1, 0, 0, () => ComputeConfiguration$, () => ScalingConfigurationInput$, 0, () => VpcConfig$, () => ProxyConfiguration$, 0, 0, () => TagList], 1
];
var UpdateFleetOutput$ = [3, n0, _UFO,
    0,
    [_fl],
    [() => Fleet$]
];
var UpdateProjectInput$ = [3, n0, _UPI,
    0,
    [_na, _de, _so, _sS, _sV, _sSV, _ar, _sA, _ca, _en, _sR, _tIM, _qTIM, _eK, _ta, _vC, _bE, _lCo, _fSL, _bBC, _cBL, _aRL],
    [0, 0, () => ProjectSource$, () => ProjectSources, 0, () => ProjectSecondarySourceVersions, () => ProjectArtifacts$, () => ProjectArtifactsList, () => ProjectCache$, () => ProjectEnvironment$, 0, 1, 1, 0, () => TagList, () => VpcConfig$, 2, () => LogsConfig$, () => ProjectFileSystemLocations, () => ProjectBuildBatchConfig$, 1, 1], 1
];
var UpdateProjectOutput$ = [3, n0, _UPO,
    0,
    [_pr],
    [() => Project$]
];
var UpdateProjectVisibilityInput$ = [3, n0, _UPVI,
    0,
    [_pAr, _pV, _rAR],
    [0, 0, 0], 2
];
var UpdateProjectVisibilityOutput$ = [3, n0, _UPVO,
    0,
    [_pAr, _pPA, _pV],
    [0, 0, 0]
];
var UpdateReportGroupInput$ = [3, n0, _URGI,
    0,
    [_a, _eCx, _ta],
    [0, () => ReportExportConfig$, () => TagList], 1
];
var UpdateReportGroupOutput$ = [3, n0, _URGO,
    0,
    [_rGe],
    [() => ReportGroup$]
];
var UpdateWebhookInput$ = [3, n0, _UWI,
    0,
    [_pN, _bF, _rS, _fG, _bT, _pRBP],
    [0, 0, 2, () => FilterGroups, 0, () => PullRequestBuildPolicy$], 1
];
var UpdateWebhookOutput$ = [3, n0, _UWO,
    0,
    [_w],
    [() => Webhook$]
];
var VpcConfig$ = [3, n0, _VC,
    0,
    [_vI, _su, _sGI],
    [0, 64 | 0, 64 | 0]
];
var Webhook$ = [3, n0, _W,
    0,
    [_ur, _pU, _se, _bF, _fG, _bT, _mC, _lMS, _sCco, _st, _sM, _pRBP],
    [0, 0, 0, 0, () => FilterGroups, 0, 2, 4, () => ScopeConfiguration$, 0, 0, () => PullRequestBuildPolicy$]
];
var WebhookFilter$ = [3, n0, _WF,
    0,
    [_t, _pat, _eMP],
    [0, 0, 2], 2
];
var CodeBuildServiceException$ = [-3, _sm, "CodeBuildServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(CodeBuildServiceException$, CodeBuildServiceException);
var BuildArtifactsList = [1, n0, _BAL,
    0, () => BuildArtifacts$
];
var BuildBatches = [1, n0, _BBu,
    0, () => BuildBatch$
];
var BuildBatchPhases = [1, n0, _BBPu,
    0, () => BuildBatchPhase$
];
var BuildGroups = [1, n0, _BGu,
    0, () => BuildGroup$
];
var BuildPhases = [1, n0, _BPu,
    0, () => BuildPhase$
];
var Builds = [1, n0, _Bu,
    0, () => Build$
];
var BuildsNotDeleted = [1, n0, _BNDu,
    0, () => BuildNotDeleted$
];
var BuildSummaries = [1, n0, _BSu,
    0, () => BuildSummary$
];
var CodeCoverages = [1, n0, _CCod,
    0, () => CodeCoverage$
];
var CommandExecutions = [1, n0, _CEo,
    0, [() => CommandExecution$,
        0]
];
var EnvironmentImages = [1, n0, _EIn,
    0, () => EnvironmentImage$
];
var EnvironmentLanguages = [1, n0, _ELn,
    0, () => EnvironmentLanguage$
];
var EnvironmentPlatforms = [1, n0, _EPn,
    0, () => EnvironmentPlatform$
];
var EnvironmentVariables = [1, n0, _EVn,
    0, () => EnvironmentVariable$
];
var ExportedEnvironmentVariables = [1, n0, _EEVx,
    0, () => ExportedEnvironmentVariable$
];
var FilterGroup = [1, n0, _FGi,
    0, () => WebhookFilter$
];
var FilterGroups = [1, n0, _FG,
    0, () => FilterGroup
];
var FleetProxyRules = [1, n0, _FPRl,
    0, () => FleetProxyRule$
];
var Fleets = [1, n0, _Fl,
    0, () => Fleet$
];
var PhaseContexts = [1, n0, _PCh,
    0, () => PhaseContext$
];
var ProjectArtifactsList = [1, n0, _PAL,
    0, () => ProjectArtifacts$
];
var ProjectFileSystemLocations = [1, n0, _PFSLr,
    0, () => ProjectFileSystemLocation$
];
var Projects = [1, n0, _Pr,
    0, () => Project$
];
var ProjectSecondarySourceVersions = [1, n0, _PSSV,
    0, () => ProjectSourceVersion$
];
var ProjectSources = [1, n0, _PSr,
    0, () => ProjectSource$
];
var ReportGroups = [1, n0, _RGe,
    0, () => ReportGroup$
];
var ReportGroupTrendRawDataList = [1, n0, _RGTRDL,
    0, () => ReportWithRawData$
];
var Reports = [1, n0, _Re,
    0, () => Report$
];
var ResolvedSecondaryArtifacts = [1, n0, _RSA,
    0, () => ResolvedArtifact$
];
var Sandboxes = [1, n0, _Sa,
    0, () => Sandbox$
];
var SandboxSessionPhases = [1, n0, _SSPa,
    0, () => SandboxSessionPhase$
];
var SourceCredentialsInfos = [1, n0, _SCIou,
    0, () => SourceCredentialsInfo$
];
var TagList = [1, n0, _TL,
    0, () => Tag$
];
var TargetTrackingScalingConfigurations = [1, n0, _TTSCa,
    0, () => TargetTrackingScalingConfiguration$
];
var TestCases = [1, n0, _TCe,
    0, () => TestCase$
];
var BatchDeleteBuilds$ = [9, n0, _BDB,
    0, () => BatchDeleteBuildsInput$, () => BatchDeleteBuildsOutput$
];
var BatchGetBuildBatches$ = [9, n0, _BGBB,
    0, () => BatchGetBuildBatchesInput$, () => BatchGetBuildBatchesOutput$
];
var BatchGetBuilds$ = [9, n0, _BGB,
    0, () => BatchGetBuildsInput$, () => BatchGetBuildsOutput$
];
var BatchGetCommandExecutions$ = [9, n0, _BGCE,
    0, () => BatchGetCommandExecutionsInput$, () => BatchGetCommandExecutionsOutput$
];
var BatchGetFleets$ = [9, n0, _BGF,
    0, () => BatchGetFleetsInput$, () => BatchGetFleetsOutput$
];
var BatchGetProjects$ = [9, n0, _BGP,
    0, () => BatchGetProjectsInput$, () => BatchGetProjectsOutput$
];
var BatchGetReportGroups$ = [9, n0, _BGRG,
    0, () => BatchGetReportGroupsInput$, () => BatchGetReportGroupsOutput$
];
var BatchGetReports$ = [9, n0, _BGR,
    0, () => BatchGetReportsInput$, () => BatchGetReportsOutput$
];
var BatchGetSandboxes$ = [9, n0, _BGS,
    0, () => BatchGetSandboxesInput$, () => BatchGetSandboxesOutput$
];
var CreateFleet$ = [9, n0, _CF,
    0, () => CreateFleetInput$, () => CreateFleetOutput$
];
var CreateProject$ = [9, n0, _CP,
    0, () => CreateProjectInput$, () => CreateProjectOutput$
];
var CreateReportGroup$ = [9, n0, _CRG,
    0, () => CreateReportGroupInput$, () => CreateReportGroupOutput$
];
var CreateWebhook$ = [9, n0, _CW,
    0, () => CreateWebhookInput$, () => CreateWebhookOutput$
];
var DeleteBuildBatch$ = [9, n0, _DBB,
    0, () => DeleteBuildBatchInput$, () => DeleteBuildBatchOutput$
];
var DeleteFleet$ = [9, n0, _DF,
    0, () => DeleteFleetInput$, () => DeleteFleetOutput$
];
var DeleteProject$ = [9, n0, _DP,
    0, () => DeleteProjectInput$, () => DeleteProjectOutput$
];
var DeleteReport$ = [9, n0, _DR,
    0, () => DeleteReportInput$, () => DeleteReportOutput$
];
var DeleteReportGroup$ = [9, n0, _DRG,
    0, () => DeleteReportGroupInput$, () => DeleteReportGroupOutput$
];
var DeleteResourcePolicy$ = [9, n0, _DRP,
    0, () => DeleteResourcePolicyInput$, () => DeleteResourcePolicyOutput$
];
var DeleteSourceCredentials$ = [9, n0, _DSC,
    0, () => DeleteSourceCredentialsInput$, () => DeleteSourceCredentialsOutput$
];
var DeleteWebhook$ = [9, n0, _DW,
    0, () => DeleteWebhookInput$, () => DeleteWebhookOutput$
];
var DescribeCodeCoverages$ = [9, n0, _DCC,
    0, () => DescribeCodeCoveragesInput$, () => DescribeCodeCoveragesOutput$
];
var DescribeTestCases$ = [9, n0, _DTC,
    0, () => DescribeTestCasesInput$, () => DescribeTestCasesOutput$
];
var GetReportGroupTrend$ = [9, n0, _GRGT,
    0, () => GetReportGroupTrendInput$, () => GetReportGroupTrendOutput$
];
var GetResourcePolicy$ = [9, n0, _GRP,
    0, () => GetResourcePolicyInput$, () => GetResourcePolicyOutput$
];
var ImportSourceCredentials$ = [9, n0, _ISC,
    0, () => ImportSourceCredentialsInput$, () => ImportSourceCredentialsOutput$
];
var InvalidateProjectCache$ = [9, n0, _IPC,
    0, () => InvalidateProjectCacheInput$, () => InvalidateProjectCacheOutput$
];
var ListBuildBatches$ = [9, n0, _LBB,
    0, () => ListBuildBatchesInput$, () => ListBuildBatchesOutput$
];
var ListBuildBatchesForProject$ = [9, n0, _LBBFP,
    0, () => ListBuildBatchesForProjectInput$, () => ListBuildBatchesForProjectOutput$
];
var ListBuilds$ = [9, n0, _LB,
    0, () => ListBuildsInput$, () => ListBuildsOutput$
];
var ListBuildsForProject$ = [9, n0, _LBFP,
    0, () => ListBuildsForProjectInput$, () => ListBuildsForProjectOutput$
];
var ListCommandExecutionsForSandbox$ = [9, n0, _LCEFS,
    0, () => ListCommandExecutionsForSandboxInput$, () => ListCommandExecutionsForSandboxOutput$
];
var ListCuratedEnvironmentImages$ = [9, n0, _LCEI,
    0, () => ListCuratedEnvironmentImagesInput$, () => ListCuratedEnvironmentImagesOutput$
];
var ListFleets$ = [9, n0, _LF,
    0, () => ListFleetsInput$, () => ListFleetsOutput$
];
var ListProjects$ = [9, n0, _LP,
    0, () => ListProjectsInput$, () => ListProjectsOutput$
];
var ListReportGroups$ = [9, n0, _LRG,
    0, () => ListReportGroupsInput$, () => ListReportGroupsOutput$
];
var ListReports$ = [9, n0, _LR,
    0, () => ListReportsInput$, () => ListReportsOutput$
];
var ListReportsForReportGroup$ = [9, n0, _LRFRG,
    0, () => ListReportsForReportGroupInput$, () => ListReportsForReportGroupOutput$
];
var ListSandboxes$ = [9, n0, _LS,
    0, () => ListSandboxesInput$, () => ListSandboxesOutput$
];
var ListSandboxesForProject$ = [9, n0, _LSFP,
    0, () => ListSandboxesForProjectInput$, () => ListSandboxesForProjectOutput$
];
var ListSharedProjects$ = [9, n0, _LSP,
    0, () => ListSharedProjectsInput$, () => ListSharedProjectsOutput$
];
var ListSharedReportGroups$ = [9, n0, _LSRG,
    0, () => ListSharedReportGroupsInput$, () => ListSharedReportGroupsOutput$
];
var ListSourceCredentials$ = [9, n0, _LSC,
    0, () => ListSourceCredentialsInput$, () => ListSourceCredentialsOutput$
];
var PutResourcePolicy$ = [9, n0, _PRP,
    0, () => PutResourcePolicyInput$, () => PutResourcePolicyOutput$
];
var RetryBuild$ = [9, n0, _RB,
    0, () => RetryBuildInput$, () => RetryBuildOutput$
];
var RetryBuildBatch$ = [9, n0, _RBB,
    0, () => RetryBuildBatchInput$, () => RetryBuildBatchOutput$
];
var StartBuild$ = [9, n0, _SB,
    0, () => StartBuildInput$, () => StartBuildOutput$
];
var StartBuildBatch$ = [9, n0, _SBB,
    0, () => StartBuildBatchInput$, () => StartBuildBatchOutput$
];
var StartCommandExecution$ = [9, n0, _SCE,
    0, () => StartCommandExecutionInput$, () => StartCommandExecutionOutput$
];
var StartSandbox$ = [9, n0, _SSt,
    0, () => StartSandboxInput$, () => StartSandboxOutput$
];
var StartSandboxConnection$ = [9, n0, _SSC,
    0, () => StartSandboxConnectionInput$, () => StartSandboxConnectionOutput$
];
var StopBuild$ = [9, n0, _SBt,
    0, () => StopBuildInput$, () => StopBuildOutput$
];
var StopBuildBatch$ = [9, n0, _SBBt,
    0, () => StopBuildBatchInput$, () => StopBuildBatchOutput$
];
var StopSandbox$ = [9, n0, _SSto,
    0, () => StopSandboxInput$, () => StopSandboxOutput$
];
var UpdateFleet$ = [9, n0, _UF,
    0, () => UpdateFleetInput$, () => UpdateFleetOutput$
];
var UpdateProject$ = [9, n0, _UP,
    0, () => UpdateProjectInput$, () => UpdateProjectOutput$
];
var UpdateProjectVisibility$ = [9, n0, _UPV,
    0, () => UpdateProjectVisibilityInput$, () => UpdateProjectVisibilityOutput$
];
var UpdateReportGroup$ = [9, n0, _URG,
    0, () => UpdateReportGroupInput$, () => UpdateReportGroupOutput$
];
var UpdateWebhook$ = [9, n0, _UW,
    0, () => UpdateWebhookInput$, () => UpdateWebhookOutput$
];

class BatchDeleteBuildsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "BatchDeleteBuilds", {})
    .n("CodeBuildClient", "BatchDeleteBuildsCommand")
    .sc(BatchDeleteBuilds$)
    .build() {
}

class BatchGetBuildBatchesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "BatchGetBuildBatches", {})
    .n("CodeBuildClient", "BatchGetBuildBatchesCommand")
    .sc(BatchGetBuildBatches$)
    .build() {
}

class BatchGetBuildsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "BatchGetBuilds", {})
    .n("CodeBuildClient", "BatchGetBuildsCommand")
    .sc(BatchGetBuilds$)
    .build() {
}

class BatchGetCommandExecutionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "BatchGetCommandExecutions", {})
    .n("CodeBuildClient", "BatchGetCommandExecutionsCommand")
    .sc(BatchGetCommandExecutions$)
    .build() {
}

class BatchGetFleetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "BatchGetFleets", {})
    .n("CodeBuildClient", "BatchGetFleetsCommand")
    .sc(BatchGetFleets$)
    .build() {
}

class BatchGetProjectsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "BatchGetProjects", {})
    .n("CodeBuildClient", "BatchGetProjectsCommand")
    .sc(BatchGetProjects$)
    .build() {
}

class BatchGetReportGroupsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "BatchGetReportGroups", {})
    .n("CodeBuildClient", "BatchGetReportGroupsCommand")
    .sc(BatchGetReportGroups$)
    .build() {
}

class BatchGetReportsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "BatchGetReports", {})
    .n("CodeBuildClient", "BatchGetReportsCommand")
    .sc(BatchGetReports$)
    .build() {
}

class BatchGetSandboxesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "BatchGetSandboxes", {})
    .n("CodeBuildClient", "BatchGetSandboxesCommand")
    .sc(BatchGetSandboxes$)
    .build() {
}

class CreateFleetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "CreateFleet", {})
    .n("CodeBuildClient", "CreateFleetCommand")
    .sc(CreateFleet$)
    .build() {
}

class CreateProjectCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "CreateProject", {})
    .n("CodeBuildClient", "CreateProjectCommand")
    .sc(CreateProject$)
    .build() {
}

class CreateReportGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "CreateReportGroup", {})
    .n("CodeBuildClient", "CreateReportGroupCommand")
    .sc(CreateReportGroup$)
    .build() {
}

class CreateWebhookCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "CreateWebhook", {})
    .n("CodeBuildClient", "CreateWebhookCommand")
    .sc(CreateWebhook$)
    .build() {
}

class DeleteBuildBatchCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "DeleteBuildBatch", {})
    .n("CodeBuildClient", "DeleteBuildBatchCommand")
    .sc(DeleteBuildBatch$)
    .build() {
}

class DeleteFleetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "DeleteFleet", {})
    .n("CodeBuildClient", "DeleteFleetCommand")
    .sc(DeleteFleet$)
    .build() {
}

class DeleteProjectCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "DeleteProject", {})
    .n("CodeBuildClient", "DeleteProjectCommand")
    .sc(DeleteProject$)
    .build() {
}

class DeleteReportCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "DeleteReport", {})
    .n("CodeBuildClient", "DeleteReportCommand")
    .sc(DeleteReport$)
    .build() {
}

class DeleteReportGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "DeleteReportGroup", {})
    .n("CodeBuildClient", "DeleteReportGroupCommand")
    .sc(DeleteReportGroup$)
    .build() {
}

class DeleteResourcePolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "DeleteResourcePolicy", {})
    .n("CodeBuildClient", "DeleteResourcePolicyCommand")
    .sc(DeleteResourcePolicy$)
    .build() {
}

class DeleteSourceCredentialsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "DeleteSourceCredentials", {})
    .n("CodeBuildClient", "DeleteSourceCredentialsCommand")
    .sc(DeleteSourceCredentials$)
    .build() {
}

class DeleteWebhookCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "DeleteWebhook", {})
    .n("CodeBuildClient", "DeleteWebhookCommand")
    .sc(DeleteWebhook$)
    .build() {
}

class DescribeCodeCoveragesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "DescribeCodeCoverages", {})
    .n("CodeBuildClient", "DescribeCodeCoveragesCommand")
    .sc(DescribeCodeCoverages$)
    .build() {
}

class DescribeTestCasesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "DescribeTestCases", {})
    .n("CodeBuildClient", "DescribeTestCasesCommand")
    .sc(DescribeTestCases$)
    .build() {
}

class GetReportGroupTrendCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "GetReportGroupTrend", {})
    .n("CodeBuildClient", "GetReportGroupTrendCommand")
    .sc(GetReportGroupTrend$)
    .build() {
}

class GetResourcePolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "GetResourcePolicy", {})
    .n("CodeBuildClient", "GetResourcePolicyCommand")
    .sc(GetResourcePolicy$)
    .build() {
}

class ImportSourceCredentialsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ImportSourceCredentials", {})
    .n("CodeBuildClient", "ImportSourceCredentialsCommand")
    .sc(ImportSourceCredentials$)
    .build() {
}

class InvalidateProjectCacheCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "InvalidateProjectCache", {})
    .n("CodeBuildClient", "InvalidateProjectCacheCommand")
    .sc(InvalidateProjectCache$)
    .build() {
}

class ListBuildBatchesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListBuildBatches", {})
    .n("CodeBuildClient", "ListBuildBatchesCommand")
    .sc(ListBuildBatches$)
    .build() {
}

class ListBuildBatchesForProjectCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListBuildBatchesForProject", {})
    .n("CodeBuildClient", "ListBuildBatchesForProjectCommand")
    .sc(ListBuildBatchesForProject$)
    .build() {
}

class ListBuildsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListBuilds", {})
    .n("CodeBuildClient", "ListBuildsCommand")
    .sc(ListBuilds$)
    .build() {
}

class ListBuildsForProjectCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListBuildsForProject", {})
    .n("CodeBuildClient", "ListBuildsForProjectCommand")
    .sc(ListBuildsForProject$)
    .build() {
}

class ListCommandExecutionsForSandboxCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListCommandExecutionsForSandbox", {})
    .n("CodeBuildClient", "ListCommandExecutionsForSandboxCommand")
    .sc(ListCommandExecutionsForSandbox$)
    .build() {
}

class ListCuratedEnvironmentImagesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListCuratedEnvironmentImages", {})
    .n("CodeBuildClient", "ListCuratedEnvironmentImagesCommand")
    .sc(ListCuratedEnvironmentImages$)
    .build() {
}

class ListFleetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListFleets", {})
    .n("CodeBuildClient", "ListFleetsCommand")
    .sc(ListFleets$)
    .build() {
}

class ListProjectsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListProjects", {})
    .n("CodeBuildClient", "ListProjectsCommand")
    .sc(ListProjects$)
    .build() {
}

class ListReportGroupsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListReportGroups", {})
    .n("CodeBuildClient", "ListReportGroupsCommand")
    .sc(ListReportGroups$)
    .build() {
}

class ListReportsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListReports", {})
    .n("CodeBuildClient", "ListReportsCommand")
    .sc(ListReports$)
    .build() {
}

class ListReportsForReportGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListReportsForReportGroup", {})
    .n("CodeBuildClient", "ListReportsForReportGroupCommand")
    .sc(ListReportsForReportGroup$)
    .build() {
}

class ListSandboxesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListSandboxes", {})
    .n("CodeBuildClient", "ListSandboxesCommand")
    .sc(ListSandboxes$)
    .build() {
}

class ListSandboxesForProjectCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListSandboxesForProject", {})
    .n("CodeBuildClient", "ListSandboxesForProjectCommand")
    .sc(ListSandboxesForProject$)
    .build() {
}

class ListSharedProjectsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListSharedProjects", {})
    .n("CodeBuildClient", "ListSharedProjectsCommand")
    .sc(ListSharedProjects$)
    .build() {
}

class ListSharedReportGroupsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListSharedReportGroups", {})
    .n("CodeBuildClient", "ListSharedReportGroupsCommand")
    .sc(ListSharedReportGroups$)
    .build() {
}

class ListSourceCredentialsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "ListSourceCredentials", {})
    .n("CodeBuildClient", "ListSourceCredentialsCommand")
    .sc(ListSourceCredentials$)
    .build() {
}

class PutResourcePolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "PutResourcePolicy", {})
    .n("CodeBuildClient", "PutResourcePolicyCommand")
    .sc(PutResourcePolicy$)
    .build() {
}

class RetryBuildBatchCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "RetryBuildBatch", {})
    .n("CodeBuildClient", "RetryBuildBatchCommand")
    .sc(RetryBuildBatch$)
    .build() {
}

class RetryBuildCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "RetryBuild", {})
    .n("CodeBuildClient", "RetryBuildCommand")
    .sc(RetryBuild$)
    .build() {
}

class StartBuildBatchCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "StartBuildBatch", {})
    .n("CodeBuildClient", "StartBuildBatchCommand")
    .sc(StartBuildBatch$)
    .build() {
}

class StartBuildCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "StartBuild", {})
    .n("CodeBuildClient", "StartBuildCommand")
    .sc(StartBuild$)
    .build() {
}

class StartCommandExecutionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "StartCommandExecution", {})
    .n("CodeBuildClient", "StartCommandExecutionCommand")
    .sc(StartCommandExecution$)
    .build() {
}

class StartSandboxCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "StartSandbox", {})
    .n("CodeBuildClient", "StartSandboxCommand")
    .sc(StartSandbox$)
    .build() {
}

class StartSandboxConnectionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "StartSandboxConnection", {})
    .n("CodeBuildClient", "StartSandboxConnectionCommand")
    .sc(StartSandboxConnection$)
    .build() {
}

class StopBuildBatchCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "StopBuildBatch", {})
    .n("CodeBuildClient", "StopBuildBatchCommand")
    .sc(StopBuildBatch$)
    .build() {
}

class StopBuildCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "StopBuild", {})
    .n("CodeBuildClient", "StopBuildCommand")
    .sc(StopBuild$)
    .build() {
}

class StopSandboxCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "StopSandbox", {})
    .n("CodeBuildClient", "StopSandboxCommand")
    .sc(StopSandbox$)
    .build() {
}

class UpdateFleetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "UpdateFleet", {})
    .n("CodeBuildClient", "UpdateFleetCommand")
    .sc(UpdateFleet$)
    .build() {
}

class UpdateProjectCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "UpdateProject", {})
    .n("CodeBuildClient", "UpdateProjectCommand")
    .sc(UpdateProject$)
    .build() {
}

class UpdateProjectVisibilityCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "UpdateProjectVisibility", {})
    .n("CodeBuildClient", "UpdateProjectVisibilityCommand")
    .sc(UpdateProjectVisibility$)
    .build() {
}

class UpdateReportGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "UpdateReportGroup", {})
    .n("CodeBuildClient", "UpdateReportGroupCommand")
    .sc(UpdateReportGroup$)
    .build() {
}

class UpdateWebhookCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("CodeBuild_20161006", "UpdateWebhook", {})
    .n("CodeBuildClient", "UpdateWebhookCommand")
    .sc(UpdateWebhook$)
    .build() {
}

const commands = {
    BatchDeleteBuildsCommand,
    BatchGetBuildBatchesCommand,
    BatchGetBuildsCommand,
    BatchGetCommandExecutionsCommand,
    BatchGetFleetsCommand,
    BatchGetProjectsCommand,
    BatchGetReportGroupsCommand,
    BatchGetReportsCommand,
    BatchGetSandboxesCommand,
    CreateFleetCommand,
    CreateProjectCommand,
    CreateReportGroupCommand,
    CreateWebhookCommand,
    DeleteBuildBatchCommand,
    DeleteFleetCommand,
    DeleteProjectCommand,
    DeleteReportCommand,
    DeleteReportGroupCommand,
    DeleteResourcePolicyCommand,
    DeleteSourceCredentialsCommand,
    DeleteWebhookCommand,
    DescribeCodeCoveragesCommand,
    DescribeTestCasesCommand,
    GetReportGroupTrendCommand,
    GetResourcePolicyCommand,
    ImportSourceCredentialsCommand,
    InvalidateProjectCacheCommand,
    ListBuildBatchesCommand,
    ListBuildBatchesForProjectCommand,
    ListBuildsCommand,
    ListBuildsForProjectCommand,
    ListCommandExecutionsForSandboxCommand,
    ListCuratedEnvironmentImagesCommand,
    ListFleetsCommand,
    ListProjectsCommand,
    ListReportGroupsCommand,
    ListReportsCommand,
    ListReportsForReportGroupCommand,
    ListSandboxesCommand,
    ListSandboxesForProjectCommand,
    ListSharedProjectsCommand,
    ListSharedReportGroupsCommand,
    ListSourceCredentialsCommand,
    PutResourcePolicyCommand,
    RetryBuildCommand,
    RetryBuildBatchCommand,
    StartBuildCommand,
    StartBuildBatchCommand,
    StartCommandExecutionCommand,
    StartSandboxCommand,
    StartSandboxConnectionCommand,
    StopBuildCommand,
    StopBuildBatchCommand,
    StopSandboxCommand,
    UpdateFleetCommand,
    UpdateProjectCommand,
    UpdateProjectVisibilityCommand,
    UpdateReportGroupCommand,
    UpdateWebhookCommand,
};
class CodeBuild extends CodeBuildClient {
}
smithyClient.createAggregatedClient(commands, CodeBuild);

const paginateDescribeCodeCoverages = core.createPaginator(CodeBuildClient, DescribeCodeCoveragesCommand, "nextToken", "nextToken", "maxResults");

const paginateDescribeTestCases = core.createPaginator(CodeBuildClient, DescribeTestCasesCommand, "nextToken", "nextToken", "maxResults");

const paginateListBuildBatches = core.createPaginator(CodeBuildClient, ListBuildBatchesCommand, "nextToken", "nextToken", "maxResults");

const paginateListBuildBatchesForProject = core.createPaginator(CodeBuildClient, ListBuildBatchesForProjectCommand, "nextToken", "nextToken", "maxResults");

const paginateListBuilds = core.createPaginator(CodeBuildClient, ListBuildsCommand, "nextToken", "nextToken", "");

const paginateListBuildsForProject = core.createPaginator(CodeBuildClient, ListBuildsForProjectCommand, "nextToken", "nextToken", "");

const paginateListCommandExecutionsForSandbox = core.createPaginator(CodeBuildClient, ListCommandExecutionsForSandboxCommand, "nextToken", "nextToken", "maxResults");

const paginateListFleets = core.createPaginator(CodeBuildClient, ListFleetsCommand, "nextToken", "nextToken", "maxResults");

const paginateListProjects = core.createPaginator(CodeBuildClient, ListProjectsCommand, "nextToken", "nextToken", "");

const paginateListReportGroups = core.createPaginator(CodeBuildClient, ListReportGroupsCommand, "nextToken", "nextToken", "maxResults");

const paginateListReports = core.createPaginator(CodeBuildClient, ListReportsCommand, "nextToken", "nextToken", "maxResults");

const paginateListReportsForReportGroup = core.createPaginator(CodeBuildClient, ListReportsForReportGroupCommand, "nextToken", "nextToken", "maxResults");

const paginateListSandboxes = core.createPaginator(CodeBuildClient, ListSandboxesCommand, "nextToken", "nextToken", "maxResults");

const paginateListSandboxesForProject = core.createPaginator(CodeBuildClient, ListSandboxesForProjectCommand, "nextToken", "nextToken", "maxResults");

const paginateListSharedProjects = core.createPaginator(CodeBuildClient, ListSharedProjectsCommand, "nextToken", "nextToken", "maxResults");

const paginateListSharedReportGroups = core.createPaginator(CodeBuildClient, ListSharedReportGroupsCommand, "nextToken", "nextToken", "maxResults");

const ArtifactNamespace = {
    BUILD_ID: "BUILD_ID",
    NONE: "NONE",
};
const ArtifactPackaging = {
    NONE: "NONE",
    ZIP: "ZIP",
};
const ArtifactsType = {
    CODEPIPELINE: "CODEPIPELINE",
    NO_ARTIFACTS: "NO_ARTIFACTS",
    S3: "S3",
};
const AuthType = {
    BASIC_AUTH: "BASIC_AUTH",
    CODECONNECTIONS: "CODECONNECTIONS",
    OAUTH: "OAUTH",
    PERSONAL_ACCESS_TOKEN: "PERSONAL_ACCESS_TOKEN",
    SECRETS_MANAGER: "SECRETS_MANAGER",
};
const BucketOwnerAccess = {
    FULL: "FULL",
    NONE: "NONE",
    READ_ONLY: "READ_ONLY",
};
const BatchReportModeType = {
    REPORT_AGGREGATED_BATCH: "REPORT_AGGREGATED_BATCH",
    REPORT_INDIVIDUAL_BUILDS: "REPORT_INDIVIDUAL_BUILDS",
};
const StatusType = {
    FAILED: "FAILED",
    FAULT: "FAULT",
    IN_PROGRESS: "IN_PROGRESS",
    STOPPED: "STOPPED",
    SUCCEEDED: "SUCCEEDED",
    TIMED_OUT: "TIMED_OUT",
};
const CacheMode = {
    LOCAL_CUSTOM_CACHE: "LOCAL_CUSTOM_CACHE",
    LOCAL_DOCKER_LAYER_CACHE: "LOCAL_DOCKER_LAYER_CACHE",
    LOCAL_SOURCE_CACHE: "LOCAL_SOURCE_CACHE",
};
const CacheType = {
    LOCAL: "LOCAL",
    NO_CACHE: "NO_CACHE",
    S3: "S3",
};
const MachineType = {
    GENERAL: "GENERAL",
    NVME: "NVME",
};
const ComputeType = {
    ATTRIBUTE_BASED_COMPUTE: "ATTRIBUTE_BASED_COMPUTE",
    BUILD_GENERAL1_2XLARGE: "BUILD_GENERAL1_2XLARGE",
    BUILD_GENERAL1_LARGE: "BUILD_GENERAL1_LARGE",
    BUILD_GENERAL1_MEDIUM: "BUILD_GENERAL1_MEDIUM",
    BUILD_GENERAL1_SMALL: "BUILD_GENERAL1_SMALL",
    BUILD_GENERAL1_XLARGE: "BUILD_GENERAL1_XLARGE",
    BUILD_LAMBDA_10GB: "BUILD_LAMBDA_10GB",
    BUILD_LAMBDA_1GB: "BUILD_LAMBDA_1GB",
    BUILD_LAMBDA_2GB: "BUILD_LAMBDA_2GB",
    BUILD_LAMBDA_4GB: "BUILD_LAMBDA_4GB",
    BUILD_LAMBDA_8GB: "BUILD_LAMBDA_8GB",
    CUSTOM_INSTANCE_TYPE: "CUSTOM_INSTANCE_TYPE",
};
const EnvironmentVariableType = {
    PARAMETER_STORE: "PARAMETER_STORE",
    PLAINTEXT: "PLAINTEXT",
    SECRETS_MANAGER: "SECRETS_MANAGER",
};
const ImagePullCredentialsType = {
    CODEBUILD: "CODEBUILD",
    SERVICE_ROLE: "SERVICE_ROLE",
};
const CredentialProviderType = {
    SECRETS_MANAGER: "SECRETS_MANAGER",
};
const EnvironmentType = {
    ARM_CONTAINER: "ARM_CONTAINER",
    ARM_EC2: "ARM_EC2",
    ARM_LAMBDA_CONTAINER: "ARM_LAMBDA_CONTAINER",
    LINUX_CONTAINER: "LINUX_CONTAINER",
    LINUX_EC2: "LINUX_EC2",
    LINUX_GPU_CONTAINER: "LINUX_GPU_CONTAINER",
    LINUX_LAMBDA_CONTAINER: "LINUX_LAMBDA_CONTAINER",
    MAC_ARM: "MAC_ARM",
    WINDOWS_CONTAINER: "WINDOWS_CONTAINER",
    WINDOWS_EC2: "WINDOWS_EC2",
    WINDOWS_SERVER_2019_CONTAINER: "WINDOWS_SERVER_2019_CONTAINER",
    WINDOWS_SERVER_2022_CONTAINER: "WINDOWS_SERVER_2022_CONTAINER",
};
const FileSystemType = {
    EFS: "EFS",
};
const LogsConfigStatusType = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const BuildBatchPhaseType = {
    COMBINE_ARTIFACTS: "COMBINE_ARTIFACTS",
    DOWNLOAD_BATCHSPEC: "DOWNLOAD_BATCHSPEC",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    STOPPED: "STOPPED",
    SUBMITTED: "SUBMITTED",
    SUCCEEDED: "SUCCEEDED",
};
const SourceAuthType = {
    CODECONNECTIONS: "CODECONNECTIONS",
    OAUTH: "OAUTH",
    SECRETS_MANAGER: "SECRETS_MANAGER",
};
const SourceType = {
    BITBUCKET: "BITBUCKET",
    CODECOMMIT: "CODECOMMIT",
    CODEPIPELINE: "CODEPIPELINE",
    GITHUB: "GITHUB",
    GITHUB_ENTERPRISE: "GITHUB_ENTERPRISE",
    GITLAB: "GITLAB",
    GITLAB_SELF_MANAGED: "GITLAB_SELF_MANAGED",
    NO_SOURCE: "NO_SOURCE",
    S3: "S3",
};
const BuildPhaseType = {
    BUILD: "BUILD",
    COMPLETED: "COMPLETED",
    DOWNLOAD_SOURCE: "DOWNLOAD_SOURCE",
    FINALIZING: "FINALIZING",
    INSTALL: "INSTALL",
    POST_BUILD: "POST_BUILD",
    PRE_BUILD: "PRE_BUILD",
    PROVISIONING: "PROVISIONING",
    QUEUED: "QUEUED",
    SUBMITTED: "SUBMITTED",
    UPLOAD_ARTIFACTS: "UPLOAD_ARTIFACTS",
};
const CommandType = {
    SHELL: "SHELL",
};
const FleetOverflowBehavior = {
    ON_DEMAND: "ON_DEMAND",
    QUEUE: "QUEUE",
};
const FleetProxyRuleBehavior = {
    ALLOW_ALL: "ALLOW_ALL",
    DENY_ALL: "DENY_ALL",
};
const FleetProxyRuleEffectType = {
    ALLOW: "ALLOW",
    DENY: "DENY",
};
const FleetProxyRuleType = {
    DOMAIN: "DOMAIN",
    IP: "IP",
};
const FleetScalingType = {
    TARGET_TRACKING_SCALING: "TARGET_TRACKING_SCALING",
};
const FleetScalingMetricType = {
    FLEET_UTILIZATION_RATE: "FLEET_UTILIZATION_RATE",
};
const FleetContextCode = {
    ACTION_REQUIRED: "ACTION_REQUIRED",
    CREATE_FAILED: "CREATE_FAILED",
    INSUFFICIENT_CAPACITY: "INSUFFICIENT_CAPACITY",
    PENDING_DELETION: "PENDING_DELETION",
    UPDATE_FAILED: "UPDATE_FAILED",
};
const FleetStatusCode = {
    ACTIVE: "ACTIVE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATING: "CREATING",
    DELETING: "DELETING",
    PENDING_DELETION: "PENDING_DELETION",
    ROTATING: "ROTATING",
    UPDATE_ROLLBACK_FAILED: "UPDATE_ROLLBACK_FAILED",
    UPDATING: "UPDATING",
};
const ProjectVisibilityType = {
    PRIVATE: "PRIVATE",
    PUBLIC_READ: "PUBLIC_READ",
};
const WebhookBuildType = {
    BUILD: "BUILD",
    BUILD_BATCH: "BUILD_BATCH",
    RUNNER_BUILDKITE_BUILD: "RUNNER_BUILDKITE_BUILD",
};
const WebhookFilterType = {
    ACTOR_ACCOUNT_ID: "ACTOR_ACCOUNT_ID",
    BASE_REF: "BASE_REF",
    COMMIT_MESSAGE: "COMMIT_MESSAGE",
    EVENT: "EVENT",
    FILE_PATH: "FILE_PATH",
    HEAD_REF: "HEAD_REF",
    ORGANIZATION_NAME: "ORGANIZATION_NAME",
    RELEASE_NAME: "RELEASE_NAME",
    REPOSITORY_NAME: "REPOSITORY_NAME",
    TAG_NAME: "TAG_NAME",
    WORKFLOW_NAME: "WORKFLOW_NAME",
};
const PullRequestBuildApproverRole = {
    BITBUCKET_ADMIN: "BITBUCKET_ADMIN",
    BITBUCKET_READ: "BITBUCKET_READ",
    BITBUCKET_WRITE: "BITBUCKET_WRITE",
    GITHUB_ADMIN: "GITHUB_ADMIN",
    GITHUB_MAINTAIN: "GITHUB_MAINTAIN",
    GITHUB_READ: "GITHUB_READ",
    GITHUB_TRIAGE: "GITHUB_TRIAGE",
    GITHUB_WRITE: "GITHUB_WRITE",
    GITLAB_DEVELOPER: "GITLAB_DEVELOPER",
    GITLAB_GUEST: "GITLAB_GUEST",
    GITLAB_MAINTAINER: "GITLAB_MAINTAINER",
    GITLAB_OWNER: "GITLAB_OWNER",
    GITLAB_PLANNER: "GITLAB_PLANNER",
    GITLAB_REPORTER: "GITLAB_REPORTER",
};
const PullRequestBuildCommentApproval = {
    ALL_PULL_REQUESTS: "ALL_PULL_REQUESTS",
    DISABLED: "DISABLED",
    FORK_PULL_REQUESTS: "FORK_PULL_REQUESTS",
};
const WebhookScopeType = {
    GITHUB_GLOBAL: "GITHUB_GLOBAL",
    GITHUB_ORGANIZATION: "GITHUB_ORGANIZATION",
    GITLAB_GROUP: "GITLAB_GROUP",
};
const WebhookStatus = {
    ACTIVE: "ACTIVE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATING: "CREATING",
    DELETING: "DELETING",
};
const ReportExportConfigType = {
    NO_EXPORT: "NO_EXPORT",
    S3: "S3",
};
const ReportPackagingType = {
    NONE: "NONE",
    ZIP: "ZIP",
};
const ReportGroupStatusType = {
    ACTIVE: "ACTIVE",
    DELETING: "DELETING",
};
const ReportType = {
    CODE_COVERAGE: "CODE_COVERAGE",
    TEST: "TEST",
};
const ReportStatusType = {
    DELETING: "DELETING",
    FAILED: "FAILED",
    GENERATING: "GENERATING",
    INCOMPLETE: "INCOMPLETE",
    SUCCEEDED: "SUCCEEDED",
};
const ReportCodeCoverageSortByType = {
    FILE_PATH: "FILE_PATH",
    LINE_COVERAGE_PERCENTAGE: "LINE_COVERAGE_PERCENTAGE",
};
const SortOrderType = {
    ASCENDING: "ASCENDING",
    DESCENDING: "DESCENDING",
};
const ReportGroupTrendFieldType = {
    BRANCHES_COVERED: "BRANCHES_COVERED",
    BRANCHES_MISSED: "BRANCHES_MISSED",
    BRANCH_COVERAGE: "BRANCH_COVERAGE",
    DURATION: "DURATION",
    LINES_COVERED: "LINES_COVERED",
    LINES_MISSED: "LINES_MISSED",
    LINE_COVERAGE: "LINE_COVERAGE",
    PASS_RATE: "PASS_RATE",
    TOTAL: "TOTAL",
};
const ServerType = {
    BITBUCKET: "BITBUCKET",
    GITHUB: "GITHUB",
    GITHUB_ENTERPRISE: "GITHUB_ENTERPRISE",
    GITLAB: "GITLAB",
    GITLAB_SELF_MANAGED: "GITLAB_SELF_MANAGED",
};
const LanguageType = {
    ANDROID: "ANDROID",
    BASE: "BASE",
    DOCKER: "DOCKER",
    DOTNET: "DOTNET",
    GOLANG: "GOLANG",
    JAVA: "JAVA",
    NODE_JS: "NODE_JS",
    PHP: "PHP",
    PYTHON: "PYTHON",
    RUBY: "RUBY",
};
const PlatformType = {
    AMAZON_LINUX: "AMAZON_LINUX",
    DEBIAN: "DEBIAN",
    UBUNTU: "UBUNTU",
    WINDOWS_SERVER: "WINDOWS_SERVER",
};
const FleetSortByType = {
    CREATED_TIME: "CREATED_TIME",
    LAST_MODIFIED_TIME: "LAST_MODIFIED_TIME",
    NAME: "NAME",
};
const ProjectSortByType = {
    CREATED_TIME: "CREATED_TIME",
    LAST_MODIFIED_TIME: "LAST_MODIFIED_TIME",
    NAME: "NAME",
};
const ReportGroupSortByType = {
    CREATED_TIME: "CREATED_TIME",
    LAST_MODIFIED_TIME: "LAST_MODIFIED_TIME",
    NAME: "NAME",
};
const SharedResourceSortByType = {
    ARN: "ARN",
    MODIFIED_TIME: "MODIFIED_TIME",
};
const RetryBuildBatchType = {
    RETRY_ALL_BUILDS: "RETRY_ALL_BUILDS",
    RETRY_FAILED_BUILDS: "RETRY_FAILED_BUILDS",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.AccountLimitExceededException = AccountLimitExceededException;
exports.AccountLimitExceededException$ = AccountLimitExceededException$;
exports.AccountSuspendedException = AccountSuspendedException;
exports.AccountSuspendedException$ = AccountSuspendedException$;
exports.ArtifactNamespace = ArtifactNamespace;
exports.ArtifactPackaging = ArtifactPackaging;
exports.ArtifactsType = ArtifactsType;
exports.AuthType = AuthType;
exports.AutoRetryConfig$ = AutoRetryConfig$;
exports.BatchDeleteBuilds$ = BatchDeleteBuilds$;
exports.BatchDeleteBuildsCommand = BatchDeleteBuildsCommand;
exports.BatchDeleteBuildsInput$ = BatchDeleteBuildsInput$;
exports.BatchDeleteBuildsOutput$ = BatchDeleteBuildsOutput$;
exports.BatchGetBuildBatches$ = BatchGetBuildBatches$;
exports.BatchGetBuildBatchesCommand = BatchGetBuildBatchesCommand;
exports.BatchGetBuildBatchesInput$ = BatchGetBuildBatchesInput$;
exports.BatchGetBuildBatchesOutput$ = BatchGetBuildBatchesOutput$;
exports.BatchGetBuilds$ = BatchGetBuilds$;
exports.BatchGetBuildsCommand = BatchGetBuildsCommand;
exports.BatchGetBuildsInput$ = BatchGetBuildsInput$;
exports.BatchGetBuildsOutput$ = BatchGetBuildsOutput$;
exports.BatchGetCommandExecutions$ = BatchGetCommandExecutions$;
exports.BatchGetCommandExecutionsCommand = BatchGetCommandExecutionsCommand;
exports.BatchGetCommandExecutionsInput$ = BatchGetCommandExecutionsInput$;
exports.BatchGetCommandExecutionsOutput$ = BatchGetCommandExecutionsOutput$;
exports.BatchGetFleets$ = BatchGetFleets$;
exports.BatchGetFleetsCommand = BatchGetFleetsCommand;
exports.BatchGetFleetsInput$ = BatchGetFleetsInput$;
exports.BatchGetFleetsOutput$ = BatchGetFleetsOutput$;
exports.BatchGetProjects$ = BatchGetProjects$;
exports.BatchGetProjectsCommand = BatchGetProjectsCommand;
exports.BatchGetProjectsInput$ = BatchGetProjectsInput$;
exports.BatchGetProjectsOutput$ = BatchGetProjectsOutput$;
exports.BatchGetReportGroups$ = BatchGetReportGroups$;
exports.BatchGetReportGroupsCommand = BatchGetReportGroupsCommand;
exports.BatchGetReportGroupsInput$ = BatchGetReportGroupsInput$;
exports.BatchGetReportGroupsOutput$ = BatchGetReportGroupsOutput$;
exports.BatchGetReports$ = BatchGetReports$;
exports.BatchGetReportsCommand = BatchGetReportsCommand;
exports.BatchGetReportsInput$ = BatchGetReportsInput$;
exports.BatchGetReportsOutput$ = BatchGetReportsOutput$;
exports.BatchGetSandboxes$ = BatchGetSandboxes$;
exports.BatchGetSandboxesCommand = BatchGetSandboxesCommand;
exports.BatchGetSandboxesInput$ = BatchGetSandboxesInput$;
exports.BatchGetSandboxesOutput$ = BatchGetSandboxesOutput$;
exports.BatchReportModeType = BatchReportModeType;
exports.BatchRestrictions$ = BatchRestrictions$;
exports.BucketOwnerAccess = BucketOwnerAccess;
exports.Build$ = Build$;
exports.BuildArtifacts$ = BuildArtifacts$;
exports.BuildBatch$ = BuildBatch$;
exports.BuildBatchFilter$ = BuildBatchFilter$;
exports.BuildBatchPhase$ = BuildBatchPhase$;
exports.BuildBatchPhaseType = BuildBatchPhaseType;
exports.BuildGroup$ = BuildGroup$;
exports.BuildNotDeleted$ = BuildNotDeleted$;
exports.BuildPhase$ = BuildPhase$;
exports.BuildPhaseType = BuildPhaseType;
exports.BuildStatusConfig$ = BuildStatusConfig$;
exports.BuildSummary$ = BuildSummary$;
exports.CacheMode = CacheMode;
exports.CacheType = CacheType;
exports.CloudWatchLogsConfig$ = CloudWatchLogsConfig$;
exports.CodeBuild = CodeBuild;
exports.CodeBuildClient = CodeBuildClient;
exports.CodeBuildServiceException = CodeBuildServiceException;
exports.CodeBuildServiceException$ = CodeBuildServiceException$;
exports.CodeCoverage$ = CodeCoverage$;
exports.CodeCoverageReportSummary$ = CodeCoverageReportSummary$;
exports.CommandExecution$ = CommandExecution$;
exports.CommandType = CommandType;
exports.ComputeConfiguration$ = ComputeConfiguration$;
exports.ComputeType = ComputeType;
exports.CreateFleet$ = CreateFleet$;
exports.CreateFleetCommand = CreateFleetCommand;
exports.CreateFleetInput$ = CreateFleetInput$;
exports.CreateFleetOutput$ = CreateFleetOutput$;
exports.CreateProject$ = CreateProject$;
exports.CreateProjectCommand = CreateProjectCommand;
exports.CreateProjectInput$ = CreateProjectInput$;
exports.CreateProjectOutput$ = CreateProjectOutput$;
exports.CreateReportGroup$ = CreateReportGroup$;
exports.CreateReportGroupCommand = CreateReportGroupCommand;
exports.CreateReportGroupInput$ = CreateReportGroupInput$;
exports.CreateReportGroupOutput$ = CreateReportGroupOutput$;
exports.CreateWebhook$ = CreateWebhook$;
exports.CreateWebhookCommand = CreateWebhookCommand;
exports.CreateWebhookInput$ = CreateWebhookInput$;
exports.CreateWebhookOutput$ = CreateWebhookOutput$;
exports.CredentialProviderType = CredentialProviderType;
exports.DebugSession$ = DebugSession$;
exports.DeleteBuildBatch$ = DeleteBuildBatch$;
exports.DeleteBuildBatchCommand = DeleteBuildBatchCommand;
exports.DeleteBuildBatchInput$ = DeleteBuildBatchInput$;
exports.DeleteBuildBatchOutput$ = DeleteBuildBatchOutput$;
exports.DeleteFleet$ = DeleteFleet$;
exports.DeleteFleetCommand = DeleteFleetCommand;
exports.DeleteFleetInput$ = DeleteFleetInput$;
exports.DeleteFleetOutput$ = DeleteFleetOutput$;
exports.DeleteProject$ = DeleteProject$;
exports.DeleteProjectCommand = DeleteProjectCommand;
exports.DeleteProjectInput$ = DeleteProjectInput$;
exports.DeleteProjectOutput$ = DeleteProjectOutput$;
exports.DeleteReport$ = DeleteReport$;
exports.DeleteReportCommand = DeleteReportCommand;
exports.DeleteReportGroup$ = DeleteReportGroup$;
exports.DeleteReportGroupCommand = DeleteReportGroupCommand;
exports.DeleteReportGroupInput$ = DeleteReportGroupInput$;
exports.DeleteReportGroupOutput$ = DeleteReportGroupOutput$;
exports.DeleteReportInput$ = DeleteReportInput$;
exports.DeleteReportOutput$ = DeleteReportOutput$;
exports.DeleteResourcePolicy$ = DeleteResourcePolicy$;
exports.DeleteResourcePolicyCommand = DeleteResourcePolicyCommand;
exports.DeleteResourcePolicyInput$ = DeleteResourcePolicyInput$;
exports.DeleteResourcePolicyOutput$ = DeleteResourcePolicyOutput$;
exports.DeleteSourceCredentials$ = DeleteSourceCredentials$;
exports.DeleteSourceCredentialsCommand = DeleteSourceCredentialsCommand;
exports.DeleteSourceCredentialsInput$ = DeleteSourceCredentialsInput$;
exports.DeleteSourceCredentialsOutput$ = DeleteSourceCredentialsOutput$;
exports.DeleteWebhook$ = DeleteWebhook$;
exports.DeleteWebhookCommand = DeleteWebhookCommand;
exports.DeleteWebhookInput$ = DeleteWebhookInput$;
exports.DeleteWebhookOutput$ = DeleteWebhookOutput$;
exports.DescribeCodeCoverages$ = DescribeCodeCoverages$;
exports.DescribeCodeCoveragesCommand = DescribeCodeCoveragesCommand;
exports.DescribeCodeCoveragesInput$ = DescribeCodeCoveragesInput$;
exports.DescribeCodeCoveragesOutput$ = DescribeCodeCoveragesOutput$;
exports.DescribeTestCases$ = DescribeTestCases$;
exports.DescribeTestCasesCommand = DescribeTestCasesCommand;
exports.DescribeTestCasesInput$ = DescribeTestCasesInput$;
exports.DescribeTestCasesOutput$ = DescribeTestCasesOutput$;
exports.DockerServer$ = DockerServer$;
exports.DockerServerStatus$ = DockerServerStatus$;
exports.EnvironmentImage$ = EnvironmentImage$;
exports.EnvironmentLanguage$ = EnvironmentLanguage$;
exports.EnvironmentPlatform$ = EnvironmentPlatform$;
exports.EnvironmentType = EnvironmentType;
exports.EnvironmentVariable$ = EnvironmentVariable$;
exports.EnvironmentVariableType = EnvironmentVariableType;
exports.ExportedEnvironmentVariable$ = ExportedEnvironmentVariable$;
exports.FileSystemType = FileSystemType;
exports.Fleet$ = Fleet$;
exports.FleetContextCode = FleetContextCode;
exports.FleetOverflowBehavior = FleetOverflowBehavior;
exports.FleetProxyRule$ = FleetProxyRule$;
exports.FleetProxyRuleBehavior = FleetProxyRuleBehavior;
exports.FleetProxyRuleEffectType = FleetProxyRuleEffectType;
exports.FleetProxyRuleType = FleetProxyRuleType;
exports.FleetScalingMetricType = FleetScalingMetricType;
exports.FleetScalingType = FleetScalingType;
exports.FleetSortByType = FleetSortByType;
exports.FleetStatus$ = FleetStatus$;
exports.FleetStatusCode = FleetStatusCode;
exports.GetReportGroupTrend$ = GetReportGroupTrend$;
exports.GetReportGroupTrendCommand = GetReportGroupTrendCommand;
exports.GetReportGroupTrendInput$ = GetReportGroupTrendInput$;
exports.GetReportGroupTrendOutput$ = GetReportGroupTrendOutput$;
exports.GetResourcePolicy$ = GetResourcePolicy$;
exports.GetResourcePolicyCommand = GetResourcePolicyCommand;
exports.GetResourcePolicyInput$ = GetResourcePolicyInput$;
exports.GetResourcePolicyOutput$ = GetResourcePolicyOutput$;
exports.GitSubmodulesConfig$ = GitSubmodulesConfig$;
exports.ImagePullCredentialsType = ImagePullCredentialsType;
exports.ImportSourceCredentials$ = ImportSourceCredentials$;
exports.ImportSourceCredentialsCommand = ImportSourceCredentialsCommand;
exports.ImportSourceCredentialsInput$ = ImportSourceCredentialsInput$;
exports.ImportSourceCredentialsOutput$ = ImportSourceCredentialsOutput$;
exports.InvalidInputException = InvalidInputException;
exports.InvalidInputException$ = InvalidInputException$;
exports.InvalidateProjectCache$ = InvalidateProjectCache$;
exports.InvalidateProjectCacheCommand = InvalidateProjectCacheCommand;
exports.InvalidateProjectCacheInput$ = InvalidateProjectCacheInput$;
exports.InvalidateProjectCacheOutput$ = InvalidateProjectCacheOutput$;
exports.LanguageType = LanguageType;
exports.ListBuildBatches$ = ListBuildBatches$;
exports.ListBuildBatchesCommand = ListBuildBatchesCommand;
exports.ListBuildBatchesForProject$ = ListBuildBatchesForProject$;
exports.ListBuildBatchesForProjectCommand = ListBuildBatchesForProjectCommand;
exports.ListBuildBatchesForProjectInput$ = ListBuildBatchesForProjectInput$;
exports.ListBuildBatchesForProjectOutput$ = ListBuildBatchesForProjectOutput$;
exports.ListBuildBatchesInput$ = ListBuildBatchesInput$;
exports.ListBuildBatchesOutput$ = ListBuildBatchesOutput$;
exports.ListBuilds$ = ListBuilds$;
exports.ListBuildsCommand = ListBuildsCommand;
exports.ListBuildsForProject$ = ListBuildsForProject$;
exports.ListBuildsForProjectCommand = ListBuildsForProjectCommand;
exports.ListBuildsForProjectInput$ = ListBuildsForProjectInput$;
exports.ListBuildsForProjectOutput$ = ListBuildsForProjectOutput$;
exports.ListBuildsInput$ = ListBuildsInput$;
exports.ListBuildsOutput$ = ListBuildsOutput$;
exports.ListCommandExecutionsForSandbox$ = ListCommandExecutionsForSandbox$;
exports.ListCommandExecutionsForSandboxCommand = ListCommandExecutionsForSandboxCommand;
exports.ListCommandExecutionsForSandboxInput$ = ListCommandExecutionsForSandboxInput$;
exports.ListCommandExecutionsForSandboxOutput$ = ListCommandExecutionsForSandboxOutput$;
exports.ListCuratedEnvironmentImages$ = ListCuratedEnvironmentImages$;
exports.ListCuratedEnvironmentImagesCommand = ListCuratedEnvironmentImagesCommand;
exports.ListCuratedEnvironmentImagesInput$ = ListCuratedEnvironmentImagesInput$;
exports.ListCuratedEnvironmentImagesOutput$ = ListCuratedEnvironmentImagesOutput$;
exports.ListFleets$ = ListFleets$;
exports.ListFleetsCommand = ListFleetsCommand;
exports.ListFleetsInput$ = ListFleetsInput$;
exports.ListFleetsOutput$ = ListFleetsOutput$;
exports.ListProjects$ = ListProjects$;
exports.ListProjectsCommand = ListProjectsCommand;
exports.ListProjectsInput$ = ListProjectsInput$;
exports.ListProjectsOutput$ = ListProjectsOutput$;
exports.ListReportGroups$ = ListReportGroups$;
exports.ListReportGroupsCommand = ListReportGroupsCommand;
exports.ListReportGroupsInput$ = ListReportGroupsInput$;
exports.ListReportGroupsOutput$ = ListReportGroupsOutput$;
exports.ListReports$ = ListReports$;
exports.ListReportsCommand = ListReportsCommand;
exports.ListReportsForReportGroup$ = ListReportsForReportGroup$;
exports.ListReportsForReportGroupCommand = ListReportsForReportGroupCommand;
exports.ListReportsForReportGroupInput$ = ListReportsForReportGroupInput$;
exports.ListReportsForReportGroupOutput$ = ListReportsForReportGroupOutput$;
exports.ListReportsInput$ = ListReportsInput$;
exports.ListReportsOutput$ = ListReportsOutput$;
exports.ListSandboxes$ = ListSandboxes$;
exports.ListSandboxesCommand = ListSandboxesCommand;
exports.ListSandboxesForProject$ = ListSandboxesForProject$;
exports.ListSandboxesForProjectCommand = ListSandboxesForProjectCommand;
exports.ListSandboxesForProjectInput$ = ListSandboxesForProjectInput$;
exports.ListSandboxesForProjectOutput$ = ListSandboxesForProjectOutput$;
exports.ListSandboxesInput$ = ListSandboxesInput$;
exports.ListSandboxesOutput$ = ListSandboxesOutput$;
exports.ListSharedProjects$ = ListSharedProjects$;
exports.ListSharedProjectsCommand = ListSharedProjectsCommand;
exports.ListSharedProjectsInput$ = ListSharedProjectsInput$;
exports.ListSharedProjectsOutput$ = ListSharedProjectsOutput$;
exports.ListSharedReportGroups$ = ListSharedReportGroups$;
exports.ListSharedReportGroupsCommand = ListSharedReportGroupsCommand;
exports.ListSharedReportGroupsInput$ = ListSharedReportGroupsInput$;
exports.ListSharedReportGroupsOutput$ = ListSharedReportGroupsOutput$;
exports.ListSourceCredentials$ = ListSourceCredentials$;
exports.ListSourceCredentialsCommand = ListSourceCredentialsCommand;
exports.ListSourceCredentialsInput$ = ListSourceCredentialsInput$;
exports.ListSourceCredentialsOutput$ = ListSourceCredentialsOutput$;
exports.LogsConfig$ = LogsConfig$;
exports.LogsConfigStatusType = LogsConfigStatusType;
exports.LogsLocation$ = LogsLocation$;
exports.MachineType = MachineType;
exports.NetworkInterface$ = NetworkInterface$;
exports.OAuthProviderException = OAuthProviderException;
exports.OAuthProviderException$ = OAuthProviderException$;
exports.PhaseContext$ = PhaseContext$;
exports.PlatformType = PlatformType;
exports.Project$ = Project$;
exports.ProjectArtifacts$ = ProjectArtifacts$;
exports.ProjectBadge$ = ProjectBadge$;
exports.ProjectBuildBatchConfig$ = ProjectBuildBatchConfig$;
exports.ProjectCache$ = ProjectCache$;
exports.ProjectEnvironment$ = ProjectEnvironment$;
exports.ProjectFileSystemLocation$ = ProjectFileSystemLocation$;
exports.ProjectFleet$ = ProjectFleet$;
exports.ProjectSortByType = ProjectSortByType;
exports.ProjectSource$ = ProjectSource$;
exports.ProjectSourceVersion$ = ProjectSourceVersion$;
exports.ProjectVisibilityType = ProjectVisibilityType;
exports.ProxyConfiguration$ = ProxyConfiguration$;
exports.PullRequestBuildApproverRole = PullRequestBuildApproverRole;
exports.PullRequestBuildCommentApproval = PullRequestBuildCommentApproval;
exports.PullRequestBuildPolicy$ = PullRequestBuildPolicy$;
exports.PutResourcePolicy$ = PutResourcePolicy$;
exports.PutResourcePolicyCommand = PutResourcePolicyCommand;
exports.PutResourcePolicyInput$ = PutResourcePolicyInput$;
exports.PutResourcePolicyOutput$ = PutResourcePolicyOutput$;
exports.RegistryCredential$ = RegistryCredential$;
exports.Report$ = Report$;
exports.ReportCodeCoverageSortByType = ReportCodeCoverageSortByType;
exports.ReportExportConfig$ = ReportExportConfig$;
exports.ReportExportConfigType = ReportExportConfigType;
exports.ReportFilter$ = ReportFilter$;
exports.ReportGroup$ = ReportGroup$;
exports.ReportGroupSortByType = ReportGroupSortByType;
exports.ReportGroupStatusType = ReportGroupStatusType;
exports.ReportGroupTrendFieldType = ReportGroupTrendFieldType;
exports.ReportGroupTrendStats$ = ReportGroupTrendStats$;
exports.ReportPackagingType = ReportPackagingType;
exports.ReportStatusType = ReportStatusType;
exports.ReportType = ReportType;
exports.ReportWithRawData$ = ReportWithRawData$;
exports.ResolvedArtifact$ = ResolvedArtifact$;
exports.ResourceAlreadyExistsException = ResourceAlreadyExistsException;
exports.ResourceAlreadyExistsException$ = ResourceAlreadyExistsException$;
exports.ResourceNotFoundException = ResourceNotFoundException;
exports.ResourceNotFoundException$ = ResourceNotFoundException$;
exports.RetryBuild$ = RetryBuild$;
exports.RetryBuildBatch$ = RetryBuildBatch$;
exports.RetryBuildBatchCommand = RetryBuildBatchCommand;
exports.RetryBuildBatchInput$ = RetryBuildBatchInput$;
exports.RetryBuildBatchOutput$ = RetryBuildBatchOutput$;
exports.RetryBuildBatchType = RetryBuildBatchType;
exports.RetryBuildCommand = RetryBuildCommand;
exports.RetryBuildInput$ = RetryBuildInput$;
exports.RetryBuildOutput$ = RetryBuildOutput$;
exports.S3LogsConfig$ = S3LogsConfig$;
exports.S3ReportExportConfig$ = S3ReportExportConfig$;
exports.SSMSession$ = SSMSession$;
exports.Sandbox$ = Sandbox$;
exports.SandboxSession$ = SandboxSession$;
exports.SandboxSessionPhase$ = SandboxSessionPhase$;
exports.ScalingConfigurationInput$ = ScalingConfigurationInput$;
exports.ScalingConfigurationOutput$ = ScalingConfigurationOutput$;
exports.ScopeConfiguration$ = ScopeConfiguration$;
exports.ServerType = ServerType;
exports.SharedResourceSortByType = SharedResourceSortByType;
exports.SortOrderType = SortOrderType;
exports.SourceAuth$ = SourceAuth$;
exports.SourceAuthType = SourceAuthType;
exports.SourceCredentialsInfo$ = SourceCredentialsInfo$;
exports.SourceType = SourceType;
exports.StartBuild$ = StartBuild$;
exports.StartBuildBatch$ = StartBuildBatch$;
exports.StartBuildBatchCommand = StartBuildBatchCommand;
exports.StartBuildBatchInput$ = StartBuildBatchInput$;
exports.StartBuildBatchOutput$ = StartBuildBatchOutput$;
exports.StartBuildCommand = StartBuildCommand;
exports.StartBuildInput$ = StartBuildInput$;
exports.StartBuildOutput$ = StartBuildOutput$;
exports.StartCommandExecution$ = StartCommandExecution$;
exports.StartCommandExecutionCommand = StartCommandExecutionCommand;
exports.StartCommandExecutionInput$ = StartCommandExecutionInput$;
exports.StartCommandExecutionOutput$ = StartCommandExecutionOutput$;
exports.StartSandbox$ = StartSandbox$;
exports.StartSandboxCommand = StartSandboxCommand;
exports.StartSandboxConnection$ = StartSandboxConnection$;
exports.StartSandboxConnectionCommand = StartSandboxConnectionCommand;
exports.StartSandboxConnectionInput$ = StartSandboxConnectionInput$;
exports.StartSandboxConnectionOutput$ = StartSandboxConnectionOutput$;
exports.StartSandboxInput$ = StartSandboxInput$;
exports.StartSandboxOutput$ = StartSandboxOutput$;
exports.StatusType = StatusType;
exports.StopBuild$ = StopBuild$;
exports.StopBuildBatch$ = StopBuildBatch$;
exports.StopBuildBatchCommand = StopBuildBatchCommand;
exports.StopBuildBatchInput$ = StopBuildBatchInput$;
exports.StopBuildBatchOutput$ = StopBuildBatchOutput$;
exports.StopBuildCommand = StopBuildCommand;
exports.StopBuildInput$ = StopBuildInput$;
exports.StopBuildOutput$ = StopBuildOutput$;
exports.StopSandbox$ = StopSandbox$;
exports.StopSandboxCommand = StopSandboxCommand;
exports.StopSandboxInput$ = StopSandboxInput$;
exports.StopSandboxOutput$ = StopSandboxOutput$;
exports.Tag$ = Tag$;
exports.TargetTrackingScalingConfiguration$ = TargetTrackingScalingConfiguration$;
exports.TestCase$ = TestCase$;
exports.TestCaseFilter$ = TestCaseFilter$;
exports.TestReportSummary$ = TestReportSummary$;
exports.UpdateFleet$ = UpdateFleet$;
exports.UpdateFleetCommand = UpdateFleetCommand;
exports.UpdateFleetInput$ = UpdateFleetInput$;
exports.UpdateFleetOutput$ = UpdateFleetOutput$;
exports.UpdateProject$ = UpdateProject$;
exports.UpdateProjectCommand = UpdateProjectCommand;
exports.UpdateProjectInput$ = UpdateProjectInput$;
exports.UpdateProjectOutput$ = UpdateProjectOutput$;
exports.UpdateProjectVisibility$ = UpdateProjectVisibility$;
exports.UpdateProjectVisibilityCommand = UpdateProjectVisibilityCommand;
exports.UpdateProjectVisibilityInput$ = UpdateProjectVisibilityInput$;
exports.UpdateProjectVisibilityOutput$ = UpdateProjectVisibilityOutput$;
exports.UpdateReportGroup$ = UpdateReportGroup$;
exports.UpdateReportGroupCommand = UpdateReportGroupCommand;
exports.UpdateReportGroupInput$ = UpdateReportGroupInput$;
exports.UpdateReportGroupOutput$ = UpdateReportGroupOutput$;
exports.UpdateWebhook$ = UpdateWebhook$;
exports.UpdateWebhookCommand = UpdateWebhookCommand;
exports.UpdateWebhookInput$ = UpdateWebhookInput$;
exports.UpdateWebhookOutput$ = UpdateWebhookOutput$;
exports.VpcConfig$ = VpcConfig$;
exports.Webhook$ = Webhook$;
exports.WebhookBuildType = WebhookBuildType;
exports.WebhookFilter$ = WebhookFilter$;
exports.WebhookFilterType = WebhookFilterType;
exports.WebhookScopeType = WebhookScopeType;
exports.WebhookStatus = WebhookStatus;
exports.paginateDescribeCodeCoverages = paginateDescribeCodeCoverages;
exports.paginateDescribeTestCases = paginateDescribeTestCases;
exports.paginateListBuildBatches = paginateListBuildBatches;
exports.paginateListBuildBatchesForProject = paginateListBuildBatchesForProject;
exports.paginateListBuilds = paginateListBuilds;
exports.paginateListBuildsForProject = paginateListBuildsForProject;
exports.paginateListCommandExecutionsForSandbox = paginateListCommandExecutionsForSandbox;
exports.paginateListFleets = paginateListFleets;
exports.paginateListProjects = paginateListProjects;
exports.paginateListReportGroups = paginateListReportGroups;
exports.paginateListReports = paginateListReports;
exports.paginateListReportsForReportGroup = paginateListReportsForReportGroup;
exports.paginateListSandboxes = paginateListSandboxes;
exports.paginateListSandboxesForProject = paginateListSandboxesForProject;
exports.paginateListSharedProjects = paginateListSharedProjects;
exports.paginateListSharedReportGroups = paginateListSharedReportGroups;
